---
title: "Z_analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##chargement des librairies
```{r load-Packages, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyverse)
```
##Définition d'un espace de travail
Chaque utilisateur définira une arborescence de travail avec plusieurs dossiers contenant respectivement les fichiers vcf, les entrées BayPass, les sorties BayPass, les sorties POD et enfin un dossier recueillant les différents résultats (plots, liste de SNP...)
```{r work-space}
path_Tables <- "C:/BayPass_pipeline/Resultats/Z_analysis/"
path_res <- "C:/BayPass_pipeline/Resultats/Z_analysis/Resultats/"
```
##Importe les fichiers de profondeur moyenne en fenètre glissante
```{r Import_coverage-Full}
SVF1.cov=read.table(paste(path_Tables, "SVF1_median.SlidingCoverage", sep=""),h=T)
SVM1.cov=read.table(paste(path_Tables, "SVM1_median.SlidingCoverage", sep=""),h=T)
SV2021.cov=read.table(paste(path_Tables, "SV2021_median.SlidingCoverage", sep=""),h=T)
RGV2021.cov=read.table(paste(path_Tables, "RGV2021_median.SlidingCoverage", sep=""),h=T)
```
## Calcul des ratios
Calcule le ratio profondeur moyenne de chaque fenêtre / profondeur moyenne des autosomes pour les ♂ et ♀
Calcule le ration p♂/p♀ => si 1 = région autosomale(diploide), si 2 région sexuelle (haploide)

```{r Calcul-coverage}
SVF1.cov <- SVF1.cov %>% mutate(pSVF1 = Mean_Depth/34.08) #pondère la profondeur par la profondeur moyenne des autosomes
SVF1.cov <- SVF1.cov %>% mutate(p2SVF1 = Median/33) #pondère la profondeur mediane par la mediane des autosomes
SVM1.cov <- SVM1.cov %>% mutate(pSVM1 = Mean_Depth/37.94)
SVM1.cov <- SVM1.cov %>% mutate(p2SVM1 = Median/37) 

SV1.j <- left_join(SVF1.cov,SVM1.cov, by = c("chr", "End_Bin"))
SV1.j <- SV1.j %>% mutate(pRatio = pSVM1/pSVF1) #calcule le ratio ♂/♀ à partir des moyennes => 1= autosomal, 2=sexuel
SV1.j <- SV1.j %>% mutate(p2Ratio = p2SVM1/p2SVF1) #calcule le ratio ♂/♀ à partir des medianes => 1= autosomal, 2=sexuel
SV1.j <- SV1.j  %>% filter(!is.infinite(pRatio)) %>% filter(!is.na(pRatio)) %>% filter(pRatio > 0) #supprime les valeurs à 0 ou #NA
SV1.j <- SV1.j  %>% filter(!is.infinite(p2Ratio)) %>% filter(!is.na(p2Ratio)) %>% filter(p2Ratio > 0) #supprime les valeurs à 0 ou #NA

SV2021.cov <- SV2021.cov %>% mutate(pSV2021 = Mean_Depth/59.60) #pondère la profondeur par la profondeur moyenne des autosomes
SV2021.cov <- SV2021.cov %>% mutate(p2SV2021 = Median/59) #pondère la profondeur mediane par la mediane des autosomes
SV2021.cov <- SV2021.cov  %>% filter(!is.infinite(pSV2021)) %>% filter(!is.na(pSV2021)) %>% filter(pSV2021 > 0) #supprime les valeurs à 0 ou #NA
SV2021.cov <- SV2021.cov  %>% filter(!is.infinite(p2SV2021)) %>% filter(!is.na(p2SV2021)) %>% filter(p2SV2021 > 0) #supprime les valeurs à 0 ou #NA
RGV2021.cov <- RGV2021.cov %>% mutate(pRGV2021 = Mean_Depth/58.96)
RGV2021.cov <- RGV2021.cov %>% mutate(p2RGV2021 = Median/58)
RGV2021.cov <- RGV2021.cov  %>% filter(!is.infinite(pRGV2021)) %>% filter(!is.na(pRGV2021)) %>% filter(pRGV2021 > 0) #supprime les valeurs à 0 ou #NA
RGV2021.cov <- RGV2021.cov  %>% filter(!is.infinite(p2RGV2021)) %>% filter(!is.na(p2RGV2021)) %>% filter(p2RGV2021 > 0) #supprime les valeurs à 0 ou #NA
```
## des ratios pour tout le génome
Représentation en ligne = + facet_grid(~chr,scales = 'free_x', space = 'free_x', switch = 'x')
Représentation en grille = + facet_wrap(~chr,scales = 'free_x', strip.position =c("bottom"))
```{r Plot, fig.width=20, fig.height=10}
SVF1.p.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=pSVF1)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVF1 Windowed Mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p=", mean(SV1.j$pSVF1))) + geom_hline(yintercept=1)#+ ylim(0, 10)
SVF1.p.plot
SVF1.p2.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=p2SVF1)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVF1 Windowed Median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p2=", mean(SV1.j$p2SVF1))) + geom_hline(yintercept=1)+ ylim(0, 10)
SVF1.p2.plot

SVM1.p.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=pSVM1)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVM1 Windowed Mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p=", mean(SV1.j$pSVM1))) + geom_hline(yintercept=1)#+ ylim(0, 10)
SVM1.p.plot
SVM1.p2.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=p2SVM1)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVM1 Windowed Median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p2=", mean(SV1.j$p2SVM1))) + geom_hline(yintercept=1)+ ylim(0, 10)
SVM1.p2.plot

SV1.p.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=(pRatio))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="Male p / Female p windowed ratio", subtitle = paste("Treshold Line = 1 \n Mean Ratio p=", mean(SV1.j$pRatio))) + geom_hline(yintercept=1)+ ylim(0, 10)
SV1.p.plot
SV1.p2.plot= ggplot(data=SV1.j, aes(x=End_Bin, y=(p2Ratio))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="Male p2 / Female p2 windowed ratio", subtitle = paste("Treshold Line = 1 \n mean Ratio p2=", mean(SV1.j$p2Ratio))) + geom_hline(yintercept=1)+ ylim(0, 10)
SV1.p2.plot

sv2021.p.plot= ggplot(data=SV2021.cov, aes(x=End_Bin, y=pSV2021)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SV2021 Windowed Mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SV2021 p=", mean(SV2021.cov$pSV2021))) + geom_hline(yintercept=1)#+ ylim(0, 10)
sv2021.p.plot
SV2021.p2.plot= ggplot(data=SV2021.cov, aes(x=End_Bin, y=p2SV2021)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SV2021 Windowed Median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SV2021 p2=", mean(SV2021.cov$p2SV2021))) + geom_hline(yintercept=1)#+ ylim(0, 10)
SV2021.p2.plot

RGV2021.p.plot= ggplot(data=RGV2021.cov, aes(x=End_Bin, y=pRGV2021)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "RGV2021 Windowed Mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean RGV2021 p=", mean(RGV2021.cov$pRGV2021))) + geom_hline(yintercept=1)#+ ylim(0, 10)
RGV2021.p.plot
RGV2021.p2.plot= ggplot(data=RGV2021.cov, aes(x=End_Bin, y=p2RGV2021)) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "RGV2021 Windowed Median relative depth", subtitle = paste("Treshold Line = 1 \n Mean RGV2021 p2=", mean(RGV2021.cov$p2RGV2021))) + geom_hline(yintercept=1)+ ylim(0, 10)
RGV2021.p2.plot
```
## des ratios pour un chromosome choisi
Représentation en ligne = + facet_grid(~chr,scales = 'free_x', space = 'free_x', switch = 'x')
Représentation en grille = + facet_wrap(~chr,scales = 'free_x', strip.position =c("bottom"))
```{r Plot, fig.width=20, fig.height=10}
sub.1=subset(SV1.j, chr== "chr1")
sub.2=subset(SV2021.cov, chr== "chr1")
sub.3=subset(RGV2021.cov, chr== "chr1")

SVF1.p.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(pSVF1))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="SVF1 Windowed mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p=", mean(sub.1$pSVF1))) + geom_hline(yintercept=1)+ ylim(0, 4)
SVF1.p.chr.plot
SVF1.p2.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(p2SVF1))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="SVF1 Windowed median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVF1 p2=", mean(sub.1$p2SVF1))) + geom_hline(yintercept=1)+ ylim(0, 4)
SVF1.p2.chr.plot

SVM1.p.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(pSVM1))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVM1 Windowed mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVM1 p=", mean(sub.1$pSVM1))) + geom_hline(yintercept=1)+ ylim(0, 4)
SVM1.p.chr.plot
SVM1.p2.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(p2SVM1))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "SVM1 Windowed median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SVM1 p2=", mean(sub.1$p2SVM1))) + geom_hline(yintercept=1)+ ylim(0, 4)
SVM1.p2.chr.plot

SV1.p.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(pRatio))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="pMale1 / pFemale1 windowed p ratio", subtitle = paste("Treshold Line = 1 \n Mean Ratio p=", mean(sub.1$pRatio))) + geom_hline(yintercept=1)+ ylim(0, 4)
SV1.p.chr.plot
SV1.p2.chr.plot= ggplot(data=sub.1, aes(x=End_Bin, y=(p2Ratio))) + geom_point(aes(color=chr), alpha=0.8, size=2.5) + labs(title ="pMale1 / pFemale1 windowed p2 ratio", subtitle = paste("Treshold Line = 1 \n Mean Ratio p=", mean(sub.1$p2Ratio))) + geom_hline(yintercept=1)+ ylim(0, 4) 
SV1.p2.chr.plot

SV2021.p.chr.plot= ggplot(data=sub.2, aes(x=End_Bin, y=(pSV2021))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="SV2021 Windowed mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean SV2021 p=", mean(sub.2$pSV2021))) + geom_hline(yintercept=1)+ ylim(0, 4)
SV2021.p.chr.plot
SV2021.p2.chr.plot= ggplot(data=sub.2, aes(x=End_Bin, y=(p2SV2021))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title ="SV2021 Windowed median relative depth", subtitle = paste("Treshold Line = 1 \n Mean SV2021 p2=", mean(sub.2$p2SV2021))) + geom_hline(yintercept=1)+ ylim(0, 4)
SV2021.p2.chr.plot

RGV2021.p.chr.plot= ggplot(data=sub.3, aes(x=End_Bin, y=(pRGV2021))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "RGV2021 Windowed mean relative depth", subtitle = paste("Treshold Line = 1 \n Mean RGV2021 p=", mean(sub.3$pRGV2021))) + geom_hline(yintercept=1)+ ylim(0, 4)
RGV2021.p.chr.plot
RGV2021.p2.chr.plot= ggplot(data=sub.3, aes(x=End_Bin, y=(p2RGV2021))) + geom_point(aes(color=chr), alpha=0.8, size=1.5) + labs(title = "RGV2021 Windowed median relative depth", subtitle = paste("Treshold Line = 1 \n Mean RGV2021 p2=", mean(sub.3$p2RGV2021))) + geom_hline(yintercept=1)+ ylim(0, 4)
RGV2021.p2.chr.plot

```


```{r Import_diversity}
SVF1.pi=read.table(paste(path_Tables, "SVF1_PI_Sliding50K.windowed.pi", sep=""),h=T)
```
```{r Calcul_diversity}
SVF1.pi <- SVF1.pi %>% mutate(Percentage_Diversity = PI*100)

```
