
---
title: "BayPass Pipeline"
author: "Jérôme OLIVARES"
date: "Septembre 2021"
output:
  html_document:
    toc: yes
    toc_depth: '2'
    df_print: paged
  github_document:
    fig_width: 15
    fig_height: 10
    dev: jpeg
  pdf_document:
    toc: yes
    toc_depth: '2'
editor_options:
  chunk_output_type: inline
---

#Prérequis:
L’utilisateur devra avoir une connaissance basique du logiciel Rstudio et être capable d’écrire et lancer des scripts sur un cluster de calcul. Les commandes décrites dans ce document ont été rédigées sous Rstudio version 1.4.1106 couplé à R 64 bits version 4.0.5. avec tous les packages nécessaires à jour.
Les lignes de commandes et scripts sous Linux ont été développés dans l’environnement bash et SLURM du cluster de calcul de la plateforme GenoToul de bioinformatique (GenoToul Bioinfo). Dans le cas d’une utilisation dans un autre environnement logiciel, l’utilisateur devra probablement effectuer des adaptations du code.
La dernière version du logiciel BayPass sera téléchargée depuis l’adresse http://www1.montpellier.inra.fr/CBGP/software/baypass/download.html et décompressée dans un répertoire local par l’utilisateur.
Le terme de chromosome sera utilisé en références aux appellations de contigs, scaffold, ou chromosomes qui correspondent aux séquences nucléotidiques du génome de référence, plus ou moins mature, qui sera utilisé.
##installation des packages
L'utilisation de ce pipeline nécessite l'installation au préalable des packages ci-dessous, il appartient à l'utilisateur de les installer sur son système.

install.packages(c("poolfstat"))  
install.packages(c("RColorBrewer"))  
install.packages(c("BiocManager"))  
BiocManager::install("mixOmics")  
install.packages(c("mvtnorm"))  
install.packages(c("geigen"))  
install.packages(c("corrplot"))  
install.packages(c("ape"))  
install.packages(c("VennDiagram"))  
install.packages(c("gridExtra"))  
install.packages(c("nVennR"))  
install.packages(c("tidyverse"))  

##chargement des librairies
```{r load-Packages, message=FALSE, warning=FALSE}
library(poolfstat) #filtrage et formatage du vcf en entrée BayPass
library(RColorBrewer) #gestion couleur des heatmaps
library(mixOmics) #"cim" pour heatmap
#library(mvtnorm)
#library(geigen)
library(corrplot) #matrice de corrélation pour heat map
#library(ape)
library(VennDiagram)
#library(gridExtra)
library(nVennR)
library(tidyverse)
library('vcfR')
```
##Définition d'un espace de travail
Chaque utilisateur définira une arborescence de travail avec plusieurs dossiers contenant respectivement les fichiers vcf, les entrées BayPass, les sorties BayPass, les sorties POD et enfin un dossier recueillant les différents résultats (plots, liste de SNP...)
```{r work-space}
path_vcf <- "C:/BayPass_pipeline/vcf/"
path_input <- "C:/BayPass_pipeline/Input/13pops/"
path_out <- "C:/BayPass_pipeline/Output/13pops/"
path_POD <- "C:/BayPass_pipeline/POD/13pops/"
path_res <- "C:/BayPass_pipeline/Resultats/13pops/"
```


#Regroupement et validation des résultats
Chaque sous jeux de données analysés va produire 8 fichiers de résultats avec des extensions différentes, un premier contrôle visuel utile est de vérifier que tous les fichiers partageant la même extension soient de taille identique en kilo ou méga-octets. Des différences manifestes sont signes de problèmes lors de l'analyse (crash, disque plein…) conduisant à des fichiers incomplets.

Avant de regrouper les résultats, il faut valider l'homogénéité des analyses en comparant les matrices Ω entre elles par un indice de distance FMD, plus l'indice sera faible plus les matrices donc les analyses seront comparables.
```{r validation-matrice-omega, fig.width=24, fig.height=18}
source("C:/BayPass_pipeline/utils/baypass_utils.R")
prefix <- "13pops-chr1_STD3K_3cov.sub"
#prefix <- "13pops-chr1_RUN"
#liste et compte les matrices Ω du répertoire "path"
path_Omega <- "C:/BayPass_pipeline/Output/13pops/chr1/Omega3K/"
listMatrix <- list.files(path_Omega, pattern="mat_omega.out")
nMatrix<-length(listMatrix)
cat("Nbr matrix files =", nMatrix, "\n")

#boucle sur toutes les matrices, calcule les distances FMD en pairwise et stocke le résultat
ListFMD<-c()
for (i in 1:nMatrix) for (j in 1:nMatrix) if(i!=j) {
omegaA=as.matrix(read.table(paste(path_Omega, prefix, i,"_mat_omega.out", sep="")))
omegaB=as.matrix(read.table(paste(path_Omega, prefix, j,"_mat_omega.out", sep="")))
FMD <- fmd.dist(omegaA, omegaB)
ListFMD <- c(ListFMD,FMD)
}
pnames <- as.character(c('13-1-S', '18-84-001-S', '17-47-003-S', '17-49-001-S', '85-3-4-S', 'IT-ID3-S', '17-47-002-R', '17-53-006-R', '19-30-1-R', '44-1-R', '44-2-R', '44-3-R', 'IT-ID1-R'))
#heatmap de la dernière matrice Ω.
colnames(omegaB) <-c(pnames)
rownames(omegaB) <-c(pnames)
cor.mat=cov2cor(omegaB)
cim_color <- colorRampPalette(rev(brewer.pal(9, "Blues")))(16)
cim(cor.mat, color = cim_color, symkey = FALSE, margins = c(10, 10), title = "Correlation map based on last "~hat(Omega))
#SVD de la dernière matrice Ω.
SVD_omega<-plot.omega(omega=omegaB, pop.names=pnames, main = expression("Singular Value Decomposition of last " * ~hat(Omega)), pos=3)
SVD_omega


pheno <- as.character(c('Susceptible',   'Susceptible',  'Susceptible', 'Susceptible', 'Susceptible', 'Susceptible', 'Resistant', 'Resistant', 'Resistant', 'Resistant', 'Resistant','Resistant','Resistant'))
geo <- as.character(c('SE',   'SE',  'SO', 'NO',   'NO',  'IT', 'SO',   'NO',  'SE','NO','NO','NO','IT'))
nomsU<- as.character(c(' ',   'SE-S-02',  ' ', '',   ' ',  ' ', ' ',   ' ',  ' ',' ',' ',' ',' '))
nomsL<- as.character(c(' ',   ' ',  ' ', 'NO-S-02',   ' ',  'IT-S-01', 'SO-R-01',   'NO-R-04',  'SE-R-01','NO-R-01','NO-R-02','NO-R-03','IT-R-01'))
nomsR<- as.character(c('SE-S-01',   '',  'SO-S-01', ' ',   'NO-S-01',  ' ', ' ',   ' ',  ' ',' ',' ',' ',' '))
#noms<- as.character(c('SE-01-S',   'SE-02-S',  'SO-01-S', 'NO-02-S',   'NO-01-S',  'IT-01-S', 'SO-01-R',   'NO-04-R',  'SE-01-R','NO-01-R','NO-02-R','NO-03-R','IT-01-R'))
noms_bis<- as.character(c("01",   "02",  "01", "02",  "01", "01", "01", "04", "01", "01", "02", "03", "01"))
pos<- as.character(c('R',   'T',  'R', 'T',   'R',  'T', 'R',   'R',  'T','R','R','R','R'))
col_geo<- c("#ffc100",   "#ffc100",  "#69ff27", "#00b0f0", "#00b0f0", "#7030a0", "#69ff27", "#00b0f0",  "#ffc100","#00b0f0","#00b0f0","#00b0f0","#7030a0")
# convertit en data.frame
tab_SVD <- data.frame(sample.id = pnames,
    phenotype = factor(pheno),
    region = factor(geo),
    #noms = factor(noms),
    #noms_bis = factor(noms_bis),
    nomsUp = factor(nomsU),
    nomsLeft = factor(nomsL),
    nomsRight = factor(nomsR),
    colPop=col_geo,
    pos = factor(pos),
    PC = SVD_omega$PC,    # the first eigenvector
    eig = SVD_omega$eig,    # the second eigenvector
    VAR = SVD_omega$pcent.var,    # variance proportion for each principal component
    stringsAsFactors = FALSE)

ggplot_pca = ggplot(data=tab_SVD, aes(x=PC.1, y=PC.2,shape=phenotype)) +
  geom_point(aes(color= region), alpha=1, size=20)+scale_shape_manual(values = c('Susceptible'=79, 'Resistant'=16))+
  ggtitle(paste0("Singular Value Decomposition of the covariance (",expression("\U03A9"),") matrix for Z chromosome"))+
  scale_colour_manual(name = "region",labels = c("IT", "NO", "SE", "SO"), values = c("#7030a0", "#00b0f0", "#ffc100","#69ff27" ),guide = guide_legend(override.aes = list(pch=126)))
ggplot_pca + scale_x_continuous() + scale_y_continuous() +
  xlab(paste("PC1 (",round(tab_SVD$VAR[1], 2), "%)" , sep="")) +
  ylab(paste("PC2 (", round(tab_SVD$VAR[2], 2), "%)" , sep="")) +
  geom_text(aes(label = noms_bis),size=8)+theme(axis.title.x = element_text(size =15, face = "bold"))+
  theme(axis.title.y = element_text(size =15, face = "bold"))

##Pour chr1/publi
ggplot_pca = ggplot(data=tab_SVD, aes(x=PC.1, y=PC.2, shape=phenotype)) +
  geom_point(aes(color= region),alpha=1, size=40)+scale_shape_manual(values = c('Susceptible'=79, 'Resistant'=16)) +
  #ggtitle(paste0("Singular Value Decomposition of the covariance (",expression("\U03A9"),") matrix for Z chromosome"))+
  scale_colour_manual(name = "region",labels = c("IT", "NO", "SE", "SO"), values = c("#7030a0", "#00b0f0", "#ffc100","#69ff27" ),guide = guide_legend(override.aes = list(pch=126)))

ggplot_pca +
  scale_x_continuous(limits=c(-0.50, 0.15)) + scale_y_continuous(limits=c(-0.50, 0.5)) + xlab(paste("PC1 (",round(tab_SVD$VAR[1], 2), "%)" , sep="")) +
  theme(axis.text.x=element_text(size=45),axis.text.y=element_text(size=45)) +
  ylab(paste("PC2 (", round(tab_SVD$VAR[2], 2), "%)" , sep="")) +
  geom_text(aes(label = nomsLeft), nudge_x = -0.08,fontface = 'bold', size=15)+
  geom_text(aes(label = nomsUp), nudge_y = 0.08,fontface = 'bold', size=15)+
  geom_text(aes(label = nomsRight), nudge_x = 0.08,fontface = 'bold', size=15)+
  theme(axis.title.x = element_text(size =60, face = "bold")) +
  theme(axis.title.y = element_text(size =60, face = "bold"))

#calcule la moyenne et la sd de toutes les distances FMD
cat("FMD mean =", mean(ListFMD), "\n")
cat("FMD sd =" , sd(ListFMD), "\n")

#install.packages(c("rgl"))

library(rgl)

noms<- as.character(c('SE-01-S',   'SE-02-S',  'SO-01-S', 'NO-02-S',   'NO-01-S',  'IT-01-S', 'SO-01-R',   'NO-04-R',  'SE-01-R','NO-01-R','NO-02-R','NO-03-R','IT-01-R'))

# Effectuer l'ACP sur les données
pca <- prcomp(omegaB[, 1:13], scale. = TRUE)
# Projeter les données sur les trois premières composantes principales
pca_3d <- pca$x[, 1:3]
# Créer le graphique 3D avec rgl
plot3d(pca_3d, type = "s", col = col_geo)
# Ajouter les étiquettes de classe
text3d(pca_3d, texts = noms, adj = c(1.25, 1, -0.5))
# Afficher le graphique
rglwidget()
```
Si les réplicats sont homogènes on merge les résultats xtx, C2, Betai pour tous les répliquats, un exemple de script est disponible à l'adresse suivante:
https://github.com/Jolivares-INRAE/Download/tree/BayPass_pipeline

Si l'analyse a été découpée en plusieurs chromosome on peut concaténer simplement en créant un nouveau fichier complet avec entêtes à partir du chr1, puis on concatène à la suite les autres fichiers sans l'entête:
cat Poolseq-chr1.Results.sorted > Poolseq-complet.results 
cat Poolseq-chr2.Results.sorted |  grep -v "MRK" - >> Poolseq-complet.results 
cat Poolseq-chr3.Results.sorted |  grep -v "MRK" - >> Poolseq-complet.results 
etc...

#Evaluation de la distribution des p.values des XtX
Cette évaluation se fait sur les p.values du fichier "XtX.merged.sorted":
```{r Distribution-XtX}
source("C:/BayPass_pipeline/utils/baypass_utils.R")
XtX.chr1_5K=read.table(paste(path_out,"chr1/13pops-chr1_STD5K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_5K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr1 5K=", quantile(XtX.chr1_5K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr1 5K=", quantile(XtX.chr1_5K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")

XtX.chr1_3K=read.table(paste(path_out,"chr1/13pops-chr1_STD3K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_3K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr1 3K=", quantile(XtX.chr1_3K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_3K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr1 3K=", quantile(XtX.chr1_3K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_3K$M_XtX) ,")", "\n")

XtX.chr1_7K=read.table(paste(path_out,"chr1/13pops-chr1_STD7K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_7K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr1 7K=", quantile(XtX.chr1_7K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_7K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr1 7K=", quantile(XtX.chr1_7K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_7K$M_XtX) ,")", "\n")

#pour chr29
XtX.chr29_5K=read.table(paste(path_out,"chr29/13pops_STD5K-chr29.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr29_5K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr29 5K=", quantile(XtX.chr29_5K$M_XtX, probs=0.95), "(Max=", max(XtX.chr29_5K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr29 5K=", quantile(XtX.chr29_5K$M_XtX, probs=0.99), "(Max=", max(XtX.chr29_5K$M_XtX) ,")", "\n")

XtX.chr29_3K=read.table(paste(path_out,"chr29/13pops_STD3K-chr29.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr29_3K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr29 3K=", quantile(XtX.chr29_3K$M_XtX, probs=0.95), "(Max=", max(XtX.chr29_3K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr29 3K=", quantile(XtX.chr29_3K$M_XtX, probs=0.99), "(Max=", max(XtX.chr29_3K$M_XtX) ,")", "\n")

XtX.chr29_7K=read.table(paste(path_out,"chr29/13pops_STD7K-chr29.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr29_7K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX chr29 7K=", quantile(XtX.chr29_7K$M_XtX, probs=0.95), "(Max=", max(XtX.chr29_7K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX chr29 7K=", quantile(XtX.chr29_7K$M_XtX, probs=0.99), "(Max=", max(XtX.chr29_7K$M_XtX) ,")", "\n")

XtX.auto_5K=read.table(paste(path_out,"autosomes/13pops-auto_STD5K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_5K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX auto 5K=", quantile(XtX.chr1_5K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX auto 5K=", quantile(XtX.chr1_5K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")

XtX.auto_3K=read.table(paste(path_out,"autosomes/13pops-auto_STD3K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_5K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX auto 3K=", quantile(XtX.chr1_5K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX auto 3K=", quantile(XtX.chr1_5K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")

XtX.auto_7K=read.table(paste(path_out,"autosomes/13pops-auto_STD7K_3cov.XtX", sep=""),h=T)
hist(10**(-1*XtX.chr1_5K$log10.1.pval.),freq=F,breaks=50)
abline(h=1)
cat("Seuil 5% XtX auto 7K=", quantile(XtX.chr1_5K$M_XtX, probs=0.95), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")
cat("Seuil 1% XtX auto 7K=", quantile(XtX.chr1_5K$M_XtX, probs=0.99), "(Max=", max(XtX.chr1_5K$M_XtX) ,")", "\n")

Manplot.XtX = ggplot(data=test_3chr, aes(x=pos, y=M_XtX)) + geom_point(aes(color=chr), alpha=0.8, size=0.25) + ggtitle("Plot XtX versus position")
Manplot.XtX + scale_x_continuous() + scale_y_continuous() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())+
  #facet_grid(~chr,scales = 'free_x', space = 'free_x', switch = 'x')
  facet_wrap(~chr, scales = 'free_x', strip.position = c("bottom"))
```
Une explication de comment interpréter cet histogramme de distribution est disponible à l'adresse:
http://varianceexplained.org/statistics/interpreting-pvalue-histogram/
Si cette distribution n’est pas normale il est souhaitable de calibrer nos statistiques avec un jeu de données simulées.

#Analyses des résultats
L'analyse est faite avec les seuils calculés précédemment, si ce calcul n'a pas été fait il faudra définir ces seuil "manuellement".
```{r Import_chr1,fig.width=10, fig.height=10}

XtX.chr1.med<-XtX.chr1_3K%>%
  left_join(XtX.chr1_5K, by=c("chr","pos"))%>%
  left_join(XtX.chr1_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_XtX, M_XtX.x, M_XtX.y)
XtX.chr1.med<- XtX.chr1.med %>%
  mutate(XtX_median = apply(XtX.chr1.med[,3:5], 1, median))%>%
  select(chr,pos,XtX_median)

XtX.chr29.med<-XtX.chr29_3K%>%
  left_join(XtX.chr29_5K, by=c("chr","pos"))%>%
  left_join(XtX.chr29_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_XtX, M_XtX.x, M_XtX.y)
XtX.chr29.med<- XtX.chr29.med %>%
  mutate(XtX_median = apply(XtX.chr29.med[,3:5], 1, median))%>%
  select(chr,pos,XtX_median)

XtX.auto.med<-XtX.auto_3K%>%
  left_join(XtX.auto_5K, by=c("chr","pos"))%>%
  left_join(XtX.auto_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_XtX, M_XtX.x, M_XtX.y)
XtX.auto.med<- XtX.auto.med %>%
  mutate(XtX_median = apply(XtX.auto.med[,3:5], 1, median))%>%
  select(chr,pos,XtX_median)

#Import des BF
BF.chr1_5K.Cov1=read.table(paste(path_out, "chr1/13pops-chr1_STD5K_Betai.Cov1", sep=""),h=T)
#BF.res.Cov2=read.table(paste(path_out, "chr1/13pops-chr1_STD5K_Betai.Cov2", sep=""),h=T)
#BF.res.Cov3=read.table(paste(path_out, "chr1/13pops-chr1_STD5K_Betai.Cov3", sep=""),h=T)
BF.chr1_3K.Cov1=read.table(paste(path_out, "chr1/13pops-chr1_STD3K_Betai.Cov1", sep=""),h=T)
BF.chr1_7K.Cov1=read.table(paste(path_out, "chr1/13pops-chr1_STD7K_Betai.Cov1", sep=""),h=T)

BF.chr1.med.Cov1<-BF.chr1_3K.Cov1%>%
  left_join(BF.chr1_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.chr1_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.chr1.med.Cov1<- BF.chr1.med.Cov1 %>%
  mutate(BF_med_M = apply(BF.chr1.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,BF_med_M)

eBPis.chr1.med.Cov1<-BF.chr1_3K.Cov1%>%
  left_join(BF.chr1_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.chr1_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.chr1.med.Cov1<- eBPis.chr1.med.Cov1 %>%
  mutate(eBPis_med_M = apply(eBPis.chr1.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_M)

BF.chr29_5K.Cov1=read.table(paste(path_out, "chr29/13pops_STD5K-chr29_Betai.Cov1", sep=""),h=T)
#BF.res.Cov2=read.table(paste(path_out, "chr29/13pops-chr1_STD5K_Betai.Cov2", sep=""),h=T)
#BF.res.Cov3=read.table(paste(path_out, "chr29/13pops-chr1_STD5K_Betai.Cov3", sep=""),h=T)
BF.chr29_3K.Cov1=read.table(paste(path_out, "chr29/13pops_STD3K-chr29_Betai.Cov1", sep=""),h=T)
BF.chr29_7K.Cov1=read.table(paste(path_out, "chr29/13pops_STD7K-chr29_Betai.Cov1", sep=""),h=T)

BF.chr29.med.Cov1<-BF.chr29_3K.Cov1%>%
  left_join(BF.chr29_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.chr29_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.chr29.med.Cov1<- BF.chr29.med.Cov1 %>%
  mutate(BF_med_M = apply(BF.chr29.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,BF_med_M)

eBPis.chr29.med.Cov1<-BF.chr29_3K.Cov1%>%
  left_join(BF.chr29_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.chr29_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.chr29.med.Cov1<- eBPis.chr29.med.Cov1 %>%
  mutate(eBPis_med_M = apply(eBPis.chr29.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_M)

BF.auto_5K.Cov1=read.table(paste(path_out, "autosomes/13pops-auto_STD5K_Betai.Cov1", sep=""),h=T)
#BF.res.Cov2=read.table(paste(path_out, "autosomes/13pops-chr1_STD5K_Betai.Cov2", sep=""),h=T)
#BF.res.Cov3=read.table(paste(path_out, "autosomes/13pops-chr1_STD5K_Betai.Cov3", sep=""),h=T)
BF.auto_3K.Cov1=read.table(paste(path_out, "autosomes/13pops-auto_STD3K_Betai.Cov1", sep=""),h=T)
BF.auto_7K.Cov1=read.table(paste(path_out, "autosomes/13pops-auto_STD7K_Betai.Cov1", sep=""),h=T)

BF.auto.med.Cov1<-BF.auto_3K.Cov1%>%
  left_join(BF.auto_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.auto_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.auto.med.Cov1<- BF.auto.med.Cov1 %>%
  mutate(BF_med_M = apply(BF.auto.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,BF_med_M)

eBPis.auto.med.Cov1<-BF.auto_3K.Cov1%>%
  left_join(BF.auto_5K.Cov1, by=c("chr","pos"))%>%
  left_join(BF.auto_7K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.auto.med.Cov1<- eBPis.auto.med.Cov1 %>%
  mutate(eBPis_med_M = apply(eBPis.auto.med.Cov1[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_M)

#Import des contraste
C2.chr1.C1_3K=read.table(paste0(path_out, "chr1/13pops_chr1_STD3K.contrast-C1"),h=T)
C2.chr1.C1_5K=read.table(paste0(path_out, "chr1/13pops_chr1_STD5K.contrast-C1"),h=T)
C2.chr1.C1_7K=read.table(paste0(path_out, "chr1/13pops_chr1_STD7K.contrast-C1"),h=T)

C2.chr1.med<-C2.chr1.C1_3K%>%
  left_join(C2.chr1.C1_5K, by=c("chr","pos"))%>%
  left_join(C2.chr1.C1_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_C2, M_C2.x, M_C2.y)
C2.chr1.med<- C2.chr1.med %>%
  mutate(C2_median = apply(C2.chr1.med[,3:5], 1, median))%>%
  select(chr,pos,C2_median)

C2.chr29.C1_3K=read.table(paste0(path_out, "chr29/13pops_STD3K-chr29.contrast"),h=T)
C2.chr29.C1_5K=read.table(paste0(path_out, "chr29/13pops_STD5K-chr29.contrast"),h=T)
C2.chr29.C1_7K=read.table(paste0(path_out, "chr29/13pops_STD7K-chr29.contrast"),h=T)

C2.chr29.med<-C2.chr29.C1_3K%>%
  left_join(C2.chr29.C1_5K, by=c("chr","pos"))%>%
  left_join(C2.chr29.C1_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_C2, M_C2.x, M_C2.y)
C2.chr29.med<- C2.chr29.med %>%
  mutate(C2_median = apply(C2.chr29.med[,3:5], 1, median))%>%
  select(chr,pos,C2_median)

C2.auto.C1_3K=read.table(paste0(path_out, "autosomes/13pops_auto_STD3K.contrast-C1"),h=T)
C2.auto.C1_5K=read.table(paste0(path_out, "autosomes/13pops_auto_STD5K.contrast-C1"),h=T)
C2.auto.C1_7K=read.table(paste0(path_out, "autosomes/13pops_auto_STD7K.contrast-C1"),h=T)

C2.auto.med<-C2.auto.C1_3K%>%
  left_join(C2.auto.C1_5K, by=c("chr","pos"))%>%
  left_join(C2.auto.C1_7K, by=c("chr","pos"))%>%
  select(chr,pos,M_C2, M_C2.x, M_C2.y)
C2.auto.med<- C2.auto.med %>%
  mutate(C2_median = apply(C2.auto.med[,3:5], 1, median))%>%
  select(chr,pos,C2_median)

#Pour V15
BF.auto_5K.Cov2=read.table(paste(path_out, "autosomes/13pops-auto_STD5K_Betai.Cov2", sep=""),h=T)
BF.auto_3K.Cov2=read.table(paste(path_out, "autosomes/13pops-auto_STD3K_Betai.Cov2", sep=""),h=T)
BF.auto_7K.Cov2=read.table(paste(path_out, "autosomes/13pops-auto_STD7K_Betai.Cov2", sep=""),h=T)

BF.auto.med.Cov2<-BF.auto_3K.Cov2%>%
  left_join(BF.auto_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.auto_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.auto.med.Cov2<- BF.auto.med.Cov2 %>%
  mutate(BF_med_V15 = apply(BF.auto.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,BF_med_V15)

eBPis.auto.med.Cov2<-BF.auto_3K.Cov2%>%
  left_join(BF.auto_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.auto_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.auto.med.Cov2<- eBPis.auto.med.Cov2 %>%
  mutate(eBPis_med_V15 = apply(eBPis.auto.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_V15)

BF.chr1_5K.Cov2=read.table(paste(path_out, "chr1/13pops-chr1_STD5K_Betai.Cov2", sep=""),h=T)
BF.chr1_3K.Cov2=read.table(paste(path_out, "chr1/13pops-chr1_STD3K_Betai.Cov2", sep=""),h=T)
BF.chr1_7K.Cov2=read.table(paste(path_out, "chr1/13pops-chr1_STD7K_Betai.Cov2", sep=""),h=T)

BF.chr1.med.Cov2<-BF.chr1_3K.Cov2%>%
  left_join(BF.chr1_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.chr1_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.chr1.med.Cov2<- BF.chr1.med.Cov2 %>%
  mutate(BF_med_V15 = apply(BF.chr1.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,BF_med_V15)

eBPis.chr1.med.Cov2<-BF.chr1_3K.Cov2%>%
  left_join(BF.chr1_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.chr1_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.chr1.med.Cov2<- eBPis.chr1.med.Cov2 %>%
  mutate(eBPis_med_V15 = apply(eBPis.chr1.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_V15)

BF.chr29_5K.Cov2=read.table(paste(path_out, "chr29/13pops_STD5K-chr29_Betai.Cov2", sep=""),h=T)
BF.chr29_3K.Cov2=read.table(paste(path_out, "chr29/13pops_STD3K-chr29_Betai.Cov2", sep=""),h=T)
BF.chr29_7K.Cov2=read.table(paste(path_out, "chr29/13pops_STD7K-chr29_Betai.Cov2", sep=""),h=T)

BF.chr29.med.Cov2<-BF.chr29_3K.Cov2%>%
  left_join(BF.chr29_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.chr29_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,BF.dB., BF.dB..x, BF.dB..y)
BF.chr29.med.Cov2<- BF.chr29.med.Cov2 %>%
  mutate(BF_med_V15 = apply(BF.chr29.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,BF_med_V15)

eBPis.chr29.med.Cov2<-BF.chr29_3K.Cov2%>%
  left_join(BF.chr29_5K.Cov2, by=c("chr","pos"))%>%
  left_join(BF.chr29_7K.Cov2, by=c("chr","pos"))%>%
  select(chr,pos,eBPis, eBPis.x, eBPis.y)
eBPis.chr29.med.Cov2<- eBPis.chr29.med.Cov2 %>%
  mutate(eBPis_med_V15 = apply(eBPis.chr29.med.Cov2[,3:5], 1, median))%>%
  select(chr,pos,eBPis_med_V15)

#Table finale génome complet
full.xtx<-rbind(XtX.chr1.med,XtX.auto.med,XtX.chr29.med)
full.contrast<-rbind(C2.chr1.med, C2.auto.med, C2.chr29.med)
full.BF.Cov1<-rbind(BF.chr1.med.Cov1, BF.auto.med.Cov1, BF.chr29.med.Cov1)
full.eBPis.Cov1<-rbind(eBPis.chr1.med.Cov1, eBPis.auto.med.Cov1, eBPis.chr29.med.Cov1)
full.BF.Cov2<-rbind(BF.chr1.med.Cov2, BF.auto.med.Cov2, BF.chr29.med.Cov2)
full.eBPis.Cov2<-rbind(eBPis.chr1.med.Cov2, eBPis.auto.med.Cov2, eBPis.chr29.med.Cov2)

Wild13_median.res<-full.xtx%>%
  left_join(full.contrast, by=c("chr","pos"))%>%
  left_join(full.BF.Cov1, by=c("chr","pos"))%>%
  left_join(full.eBPis.Cov1,by=c("chr","pos"))%>%
  left_join(full.BF.Cov2, by=c("chr","pos"))%>%
  left_join(full.eBPis.Cov2,by=c("chr","pos"))%>%
  mutate(N = str_sub(chr, 4, -1)) %>%
  mutate_at(c(3:9), as.numeric)

#Save
#write.table(Wild13_median.res, file=paste(path_res, "Wild13_median.res.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)
#Load
# Wild13_median.res=read.table(paste(path_res, "Wild13_median.res.txt", sep=""),h=T)
# Wild13_median.Top1XtX<-subset(Wild13_median.res, XtX_median >= quantile(Wild13_median.res$XtX_median, probs=0.99))

All_mut<-rbind(XtX.chr1_5K, XtX.auto_5K, XtX.chr29_5K)%>%
  select(chr,pos,All1, All2)%>%
  left_join(Wild13_median.res, by=c("chr","pos"))


cat("Seuil 1% XtX =", quantile(Wild13_median.res$XtX_median, probs=0.99), "(Max=", max(Wild13_median.res$XtX_median) ,")", "\n")
cat("Seuil 1% C2 =", quantile(Wild13_median.res$C2_median, probs=0.95), "(Max=", max(Wild13_median.res$C2_median) ,")", "\n")
cat("Seuil 1% BF-STD-M =", quantile(Wild13_median.res$BF_med_M, probs=0.99), "(Max=", max(Wild13_median.res$BF_med_M) ,")", "\n")
cat("Seuil 1% BF-STD-V15 =", quantile(Wild13_median.res$BF_med_V15, probs=0.99), "(Max=", max(Wild13_median.res$BF_med_V15) ,")", "\n")

Wild13_median.Top1<-subset(All_mut, BF_med_M > quantile(All_mut$BF_med_M, probs=0.99))
Wild13_median.Down1<-subset(All_mut, BF_med_M < quantile(All_mut$BF_med_M, probs=0.01))

Wild13_median.BF15_XtX1<-subset(Wild13_median.res, Wild13_median.res$BF_med_M >=15 & XtX_median > quantile(Wild13_median.res$XtX_median, probs=0.99))

test_A<-XtX.auto_5K %>%
  left_join(BF.auto_5K.Cov1, by=c("chr","pos"))%>%
  filter(chr %in% c("chr15","chr19")) %>%
  select(chr,pos,M_P,log10.1.pval., M_XtX, BF.dB.)
test_Z<-XtX.chr1_5K %>%
  left_join(BF.chr1_5K.Cov1, by=c("chr","pos"))%>%
  select(chr,pos,M_P,log10.1.pval., M_XtX, BF.dB.)
test_3chr<-rbind(test_Z,test_A)

# write.table(Wild13_median.BF15, file=paste(path_res, "Wild13_median.BF15.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)
# write.table(Wild13_median.Top1, file=paste(path_res, "Wild13_median.Top1_BF.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)
write.table(test_3chr, file=paste(path_res, "test_3chr.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)


Manplot.res = ggplot(NULL, aes(x=XtX_median, y=BF_med_M)) +
  geom_point(data=subset(Wild13_median.res, XtX_median >= quantile(Wild13_median.res$XtX_median, probs=0.99) | BF_med_M >=15), color="red", alpha=0.8, size=1)+
  geom_point(data=subset(Wild13_median.res, XtX_median < quantile(Wild13_median.res$XtX_median, probs=0.99) | BF_med_M <15), color="blue",alpha=0.8, size=1)+
  ggtitle("XtX versus BF Phenotype M (Seuil = Top 1% XtX & BF >=15)")
Manplot.res + geom_hline(yintercept=15)+
  geom_vline(xintercept=quantile(Wild13_median.res$XtX_median, probs=0.99))

Manplot.res = ggplot(NULL, aes(x=XtX_median, y=BF_med_M)) +
  geom_point(data=subset(Wild13_median.res, XtX_median >= quantile(Wild13_median.res$XtX_median, probs=0.95) | BF_med_M >=15), color="red", alpha=0.8, size=1)+
  geom_point(data=subset(Wild13_median.res, XtX_median < quantile(Wild13_median.res$XtX_median, probs=0.95) | BF_med_M <15), color="blue",alpha=0.8, size=1)+
  ggtitle("XtX versus BF Phenotype M (Seuil = Top 5% XtX & BF >=15)")
Manplot.res + geom_hline(yintercept=15)+
  geom_vline(xintercept=quantile(Wild13_median.res$XtX_median, probs=0.95))

#Fenêtre glissantes
window_size <- 50000
step_size <- 5000
#Subset des conditions que l'on veut ploter
df<-subset(Wild13_median.res, XtX_median >= quantile(Wild13_median.res$XtX_median, probs=0.99) & BF_med_M >=15)

# create an empty data frame to store the results
results_df <- data.frame(chr = character(), pos = numeric(), Start = numeric(), End = numeric(), nValues = numeric())

# iterate over each chromosome
for (j in unique(df$chr)) {
  # subset the data to the current chromosome
  chr_data <- subset(df, chr == j)
  # iterate over each window for the current chromosome
  for (i in seq(1, max(chr_data$pos) - window_size, step_size)) {
    # subset the data to the current window
    window_data <- subset(chr_data, pos >= i & pos <= i + window_size)
    # count the number of rows in the window
    num_rows_in_window <- nrow(window_data)
    # add the results to the data frame
    results_df <- rbind(results_df, data.frame(chr = j, Start = i, End = i + window_size, nValues = num_rows_in_window))
  }
}
results_df<-results_df%>%
  mutate(N = str_sub(chr, 4, -1))%>%
  mutate_at(c(2:5), as.numeric)

#write.table(results_df, file=paste(path_res, "SlideR.BF15.xtx1.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)

```
##Plot des résultats
```{r plot, fig.width=20, fig.height=10}

##fenetre glissante linux
# Slide_BF15=read.table(paste(path_res, "Wild13_median.BF15_XtX1.sliding", sep=""),h=T)
# Slide_BF15<-Slide_BF15%>%
#   mutate(N = str_sub(chr, 4, -1))%>%
#   mutate_at(c(2:5), as.numeric)

#on plote que le top 1% XtX des résultats donc on réduit le data pour gagner dutemps
Wild13_Sub.res<-subset(Wild13_median.res, XtX_median > 20.01)

####### Plot simple du Top1% XtX + BF >=15 => PUBLI ########
#Tous chr/wrap
Manplot.Slide1 = ggplot(data = Wild13_Sub.res, aes(x = pos, y = XtX_median)) +
  geom_rect(aes(xmin = 515.10e5, xmax = 545.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 1)) +
  geom_rect(aes(xmin = 75.10e5, xmax = 86.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 19)) +
  geom_rect(aes(xmin = 99.10e5, xmax = 108.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 15)) +
  geom_rect(aes(xmin = 130.10e5, xmax = 140.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 16)) +
  geom_rect(aes(xmin = 160.10e5, xmax = 175.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 2)) +
  geom_rect(aes(xmin = 160.10e5, xmax = 180.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 3)) +
  geom_rect(aes(xmin = 115.10e5, xmax = 127.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 4)) +
  geom_rect(aes(xmin = 130.10e5, xmax = 138.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 21)) +
  geom_point(data = Wild13_Sub.res, col = 'blue', alpha = 0.8, size = 1.5) +
  geom_point(data = subset(Wild13_Sub.res,  BF_med_M >= 15), col = 'red', alpha = 0.8, size = 2.5)  

Manplot.Slide1  +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  theme(axis.title.y = element_blank(), axis.text.y = element_text(size = 15)) +
  scale_y_continuous(limits=c(20, NA)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y = "posterior mean of XtX") +
  theme(axis.title.y = element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15)) +
  theme(plot.title = element_text(size = 30)) +
  facet_wrap(~N, scales = 'free_x', strip.position = c("bottom"))

#Tous chr/grid
Manplot.Slide1 = ggplot(data = Wild13_Sub.res, aes(x = pos, y = XtX_median)) +
  geom_rect(aes(xmin = 505.10e5, xmax = 565.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 1)) +
  geom_rect(aes(xmin = 50.10e5, xmax = 110.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 19)) +
  geom_rect(aes(xmin = 70.10e5, xmax = 130.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 15)) +
  geom_rect(aes(xmin = 100.10e5, xmax = 160.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 16)) +
  geom_rect(aes(xmin = 130.10e5, xmax = 190.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 2)) +
  geom_rect(aes(xmin = 140.10e5, xmax = 200.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 3)) +
  geom_rect(aes(xmin = 90.10e5, xmax = 150.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 4)) +
  geom_rect(aes(xmin = 100.10e5, xmax = 160.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Wild13_Sub.res, N == 21)) +
  geom_point(data = Wild13_Sub.res, col = 'blue', alpha = 0.8, size = 1.5) +
  geom_point(data = subset(Wild13_Sub.res, BF_med_M >= 15), col = 'red', alpha = 0.8, size = 2.5)  

Manplot.Slide1  +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  theme(axis.title.y = element_blank(), axis.text.y = element_text(size = 15)) +
  scale_y_continuous(limits=c(20, NA)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y = "posterior mean of XtX") +
  theme(axis.title.y = element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15)) +
  theme(plot.title = element_text(size = 30)) +
  facet_grid(~N, scales = 'free_x', space = 'free_x', switch = 'x')

#Sur sélection de chr:
Subdata<-subset(Wild13_Sub.res, N %in% c(1,2,3,4,15,16,19,21))

Manplot.Slide1b = ggplot(data = Subdata ,aes(x = pos, y = XtX_median)) +
  geom_rect(aes(xmin = 520.10e5, xmax = 550.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 1)) +
  geom_rect(aes(xmin = 65.10e5, xmax = 95.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 19)) +
  geom_rect(aes(xmin = 90.10e5, xmax = 120.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 15)) +
  geom_rect(aes(xmin = 120.10e5, xmax = 150.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 16)) +
  geom_rect(aes(xmin = 155.10e5, xmax = 185.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 2)) +
  geom_rect(aes(xmin = 155.10e5, xmax = 185.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 3)) +
  geom_rect(aes(xmin = 105.10e5, xmax = 135.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 4)) +
  geom_rect(aes(xmin = 110.10e5, xmax = 140.10e5, ymin = 20, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Subdata, N == 21)) +
  geom_point(data = subset(Subdata, XtX_median > 20.01), col = 'blue', alpha = 0.8, size = 3) +
  geom_point(data = subset(Subdata, XtX_median > 20.01 & BF_med_M >= 15), col = 'red', alpha = 0.8, size = 4)  

Manplot.Slide1b  +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  theme(axis.title.y = element_blank(), axis.text.y = element_text(size = 15)) +
  scale_y_continuous(limits=c(20, NA)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y = "posterior mean of XtX") +
  theme(axis.title.y = element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15)) +
  theme(plot.title = element_text(size = 30)) +
  facet_grid(~N, scales = 'free_x', space = 'free_x', switch = 'x')





#######Plot simple des BF>15 & XtX>top1% => PUBLI ########
#Tous chr/wrap
Manplot.Slide2 = ggplot(NULL, aes(x=Start, y=BF))+
  geom_rect(aes(xmin = 525.10e5, xmax = 545.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 1)) +
  geom_rect(aes(xmin = 75.10e5, xmax = 86.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 19)) +
  geom_rect(aes(xmin = 99.10e5, xmax = 108.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 15)) +
  geom_rect(aes(xmin = 130.10e5, xmax = 140.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 16)) +
  geom_rect(aes(xmin = 160.10e5, xmax = 175.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 2)) +
  geom_rect(aes(xmin = 160.10e5, xmax = 180.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 3)) +
  geom_rect(aes(xmin = 115.10e5, xmax = 127.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 4)) +
  geom_rect(aes(xmin = 127.10e5, xmax = 135.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 21)) +
  geom_point(data=subset(Slide_BF15, BF>0 ), col='red', alpha=0.8, size=3)+
  geom_point(data=subset(Slide_BF15, chr=="chr1" & Start>=53570000 & End<=53620000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8030000 & End<=8115000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8140000 & End<=8190000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, BF==0 ), col='white', alpha=0.8, size=4) 
  #ggtitle("sliding 50K window BF median ≥15")+
Manplot.Slide2 +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
  theme(axis.title.y = element_blank(), axis.text.y = element_text(size = 15)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y = "Number of Bayes Factor >= 15") +
  theme(axis.title.y = element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15)) +
  theme(plot.title = element_text(size = 30)) +
  facet_wrap(~N, scales = 'free_x', strip.position = c("bottom"))
  
#Tous chr/grid
Manplot.Slide2b = ggplot(NULL, aes(x=Start, y=BF))+
  geom_rect(aes(xmin = 505.10e5, xmax = 565.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 1)) +
  geom_rect(aes(xmin = 50.10e5, xmax = 110.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 19)) +
  geom_rect(aes(xmin = 70.10e5, xmax = 130.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 15)) +
  geom_rect(aes(xmin = 100.10e5, xmax = 160.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 16)) +
  geom_rect(aes(xmin = 130.10e5, xmax = 190.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 2)) +
  geom_rect(aes(xmin = 140.10e5, xmax = 200.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 3)) +
  geom_rect(aes(xmin = 90.10e5, xmax = 150.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 4)) +
  geom_rect(aes(xmin = 100.10e5, xmax = 160.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 21)) +
  geom_point(data=subset(Slide_BF15, BF>0 ), col='red', alpha=0.8, size=3)+
  geom_point(data=subset(Slide_BF15, chr=="chr1" & Start>=53570000 & End<=53620000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8030000 & End<=8115000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8140000 & End<=8190000), col='green', alpha=0.8, size=4) +
  geom_point(data=subset(Slide_BF15, BF==0 ), col='white', alpha=0.8, size=4) 
  #ggtitle("sliding 50K window BF median ≥15")+

  
Manplot.Slide2b +
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) +
  theme(axis.title.y=element_blank(), axis.text.y = element_text(size = 15)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y=("Number of Bayes Factor >= 15")) +
  theme(axis.title.y=element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15))+
  facet_grid(~N, scales = 'free_x', space = 'free_x', switch = 'x')

#selection chr/grid
Manplot.Slide2c = ggplot(NULL, aes(x=Start, y=BF))+
  geom_rect(aes(xmin = 520.10e5, xmax = 550.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 1)) +
  geom_rect(aes(xmin = 65.10e5, xmax = 95.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 19)) +
  geom_rect(aes(xmin = 90.10e5, xmax = 120.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 15)) +
  geom_rect(aes(xmin = 120.10e5, xmax = 150.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 16)) +
  geom_rect(aes(xmin = 155.10e5, xmax = 185.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 2)) +
  geom_rect(aes(xmin = 155.10e5, xmax = 185.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 3)) +
  geom_rect(aes(xmin = 105.10e5, xmax = 135.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 4)) +
  geom_rect(aes(xmin = 110.10e5, xmax = 140.10e5, ymin = 0, ymax = Inf), fill = "lightgray" , alpha = 0.1, data = subset(Slide_BF15, N == 21)) +
  geom_point(data=subset(Slide_BF15, BF>0 & N %in% (c(1,2,3,4,15,16,19,21))), col='red', alpha=0.8, size=4)+
  geom_point(data=subset(Slide_BF15, chr=="chr1" & Start>=53570000 & End<=53620000), col='green', alpha=0.8, size=5) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8030000 & End<=8115000), col='green', alpha=0.8, size=5) +
  geom_point(data=subset(Slide_BF15, chr=="chr19" & Start>=8140000 & End<=8190000), col='green', alpha=0.8, size=5) +
  geom_point(data=subset(Slide_BF15, BF==0 & N %in% (c(1,2,3,4,15,16,19,21))), col='white', alpha=0.8, size=5) 
  #ggtitle("sliding 50K window BF median ≥15")+
  
Manplot.Slide2c +
  theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) +
  theme(axis.title.y=element_blank(), axis.text.y = element_text(size = 15)) +
  theme(panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major.y = element_line(color = "black")) +
  labs(y=("Number of Bayes Factor >= 15")) +
  theme(axis.title.y=element_text(size = 30)) +
  theme(strip.text.x = element_text(size = 15)) +
  facet_grid(~N, scales = 'free_x', space = 'free_x', switch = 'x')
############################################################################


# #Plot des fenêtres glissantes "R"
# Manplot.Slide2 = ggplot(NULL, aes(x=Start, y=nValues))+
#   geom_point(data=results_df, col='blue', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr1" &Start>53560000 & Start<53570000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr1" &Start>53600000 & Start<53660000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr15" & Start>10350000 & Start<13315000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr16" & Start>13255000 & Start<13315000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr16" & Start>13280000 & Start<13340000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr19" & Start>8110000 & Start<8160000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr19" & Start>8175000 & Start<8245000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr2" & Start>16555000 & Start<16720000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr21" & Start>13045000 & Start<13285000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr3" & Start>16845000 & Start<17150000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr4" & Start>12135000 & Start<12220000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr21" & Start>12260000 & Start<12355000), col='darkgreen', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr1" &Start>53570000 & Start<53600000), col='red', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr19" & Start>8160000 & Start<8175000), col='red', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr19" & Start>8013871 & Start<8094786), col='red', alpha=0.8, size=2)+
#   geom_point(data=subset(results_df, chr=="chr15" & Start>9800000 & Start<9890000), col='red', alpha=0.8, size=2)+
#   ggtitle("sliding 50K window BF median >=15 (Rouge = TP61F & IAP & smoothened) ")
# Manplot.Slide2+ theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank(),panel.background = element_blank())+
#   facet_wrap(~N, scales = 'free_x', strip.position =c("bottom"))
#   #facet_grid(~N, scales = 'free_x', space = 'free_x', switch = 'x')
#     


```
Le premier graphique est un Manhattan plot des valeurs de C2 le long des chromosomes disposés en lignes (facet_grid). Pour une disposition en grille il faudra utiliser l'option "facet_wrap": facet_wrap(~chr,scales = 'free_x', strip.position =c("bottom")).
Toutes les combinaisons sont possible (XtX/C2, C2/BF...) selon l'information que l'on souhaite en tirer.
Le deuxième graphique est un Manhattan plot selon 2 conditions: toutes les positions pour lesquelles le XtX et le BF sont supérieurs aux seuils définis
Enfin on peut exporter dans un fichier les données qui ont servies à établir le deuxième graphique.

#Comparaison par diagramme de Venn:
```{r Listes}
#Filtrage selon conditions
Datmp1 = subset(Full_M.res, BF.dB.>= 15)
Datmp2 = subset(Full_V15.res, BF.dB.>= 15)
#Datmp4 = subset(XtX_BF_Join2, M_XtX> thresh.XtX & BF.dB.>thresh.BF2)

#Conversion en liste au format "chr1_pos1, chr1_pos2, chr1_pos3, ...", 
List.SNP1<-paste(Datmp1[,1], "_", Datmp1[,2], sep="")
List.SNP2<-paste(Datmp2[,1], "_", Datmp2[,2], sep="")
List.SNP3<-paste(Datmp3[,1], "_", Datmp3[,2], sep="")
```

```{r Export}
#Exporte les données filtrées sous conditions
#write.table(Manplot.thresh$data, row.names = FALSE, file=paste(path_res, "XtX-C2_VS_pos.txt", sep=""), quote = FALSE, sep = "\t")

#créé et exporte un fichier au format bed
list.tmp <- paste(Datmp1[,1], Datmp1[,2]-1, Datmp1[,2])
List.bed<-separate(plyr::ldply(list.tmp, cbind), col = "1",  sep = " ", into = c("chr", "start", "end"))
write.table(List.bed, row.names = FALSE, file=paste(path_res, "List_Outliers_BF15-chr1_M.bed", sep=""), quote = FALSE, sep = "\t")
```

```{r venn}
#Diagramme de Venn
vd0 <- venn.diagram(x=list("." = List.SNP1, "Phenotype V15" = List.SNP2, "Phenotype M" = List.SNP1), fill = c("white", "darkgreen", "blue"), cat.col = c("white", "darkgreen", "blue"), cat.cex = 1.5, fontface = "bold", filename = NULL, main = "Overlap BF>15 phenotype M vs V15")

grid.newpage()
grid.draw(vd0)

```

#Extraire les listes de chaque intersecte du diagramme:
```{r List-Venn}
myV <- plotVenn(list("BF" = List.SNP1, "Souches" = List.SNP2, "BF2" = List.SNP1))
myV <- plyr::ldply(listVennRegions(myV), cbind)
#write.table(myV, file=paste(path_res, "Overlap_BF15.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)

#récupère la liste des intersectes
myV %>% distinct(myV[,1])
```
#Convertir les liste de SNP dans un intersecte en fichier bed
```{r Bed-Venn}
#subset en fonction de l'intersecte que l'on veut
tmp<-subset(myV,myV[,1] == "1, 1, 1 (BF, Souches, BF2)")
V.tmp<-separate(plyr::ldply(paste(tmp[,2]), cbind), col = "1",  sep = "_", into = c("chr", "pos"))
V.bed <- plyr::ldply(paste(V.tmp[,1], as.numeric(V.tmp[,2])-1, V.tmp[,2]), cbind)
V.bed<-mutate(separate(V.bed, col = "1",  sep = " ", into = c("chr", "start", "end")))
#write.table(V.bed, file=paste(path_res, "overlap_Wild-souches_chr1.bed", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)
```

#Extraction des fréquences du vcf
```{r}
var_vcf <- read.vcfR(paste(path_vcf, "Wild13pops_chr1-varscan.vcf.gz", sep =""))
#on extrait les fréquences de l'allèle alternatif
gt <- extract.gt(var_vcf, element = "FREQ", as.numeric = TRUE)

#on convertit en data.frame
Data.var1 <- as.data.frame(gt)
Data.var1 <- na.omit(Data.var1)
colnames(Data.var1)=gsub("-","_", pnames)
```
Filtrage simple des SNP et extraction des fréquences correspondantes
```{r}
Candidate = Joined.res1

#fusionner les fréquences et les resultats "Joined.res"
Data.var2 <- tibble::rownames_to_column(Data.var1, "VALUE")
Data.var2<-Data.var2 %>%  separate(VALUE, c("chr", "pos"), "_", convert = TRUE)
## a degager ?=> Joined.res.M<-left_join(Candidate,Data.var2, by=c("chr","pos"))
#Joined.res.M.BF15<-left_join(subset(Candidate,BF.dB.>15),Data.var2, by=c("chr","pos"))
Joined.res.M.BF15<-left_join(Candidate,Data.var2, by=c("chr","pos"))
Joined.res.M.BF15<-arrange(Joined.res.M.BF15, chr, pos)
#vérifier les entête de colonnes à extraire
head(Joined.res.M.BF15)
#Extraire les résultats avec les bonnes colonnes
#write.table(Joined.res.M.BF15[, c(1:4,8,16,28,32:44)], file=paste(path_res, "13pops_STD_chr1_BF15_FREQ.txt", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)
```
Extraction des SNP d'un Venn et extraction des fréquences correspondantes
```{r}
#on récupère l'intersect VENN que l'on souhaite
tmp<-subset(myV,myV[,1] == "1, 1, 1 (BF, Souches, BF2)") # 1 intersecte
#tmp<-subset(myV,myV[,1] == "1, 1, 1 (M-Aux, V15-Aux, M-Std)" | myV[,1] == "1, 0, 1 (M-Aux, M-Std)") # 2 intersecte

#on ne garde que la colonne position/SNP
tmp<-paste(tmp[,2], sep="")

#on filtre le VCF avec les SNP présent dans tmp
Data.Venn=filter(Data.var1, row.names(Data.var1) %in% tmp,)
Data.Venn <- tibble::rownames_to_column(Data.Venn, "VALUE") #convertit les noms de lignes en colonne
Data.Venn<-Data.Venn %>%  separate(VALUE, c("chr", "pos"), "_", convert = TRUE)
#Joined.res1.M.BF15<-right_join(subset(Joined.res1,BF.dB.>thresh.BF), Data.Venn, by=c("chr","pos")) #avec filtre
Joined.res1.M.BF15<-right_join(Joined.res1, Data.Venn, by=c("chr","pos")) #sans filtre
Joined.res1.M.BF15<-arrange(Joined.res1.M.BF15, chr, pos)


#write.table(Joined.res1.M.BF15[, c(1:4,8,16,28,32:44)], file=paste(path_res, "Venn_Candidate_wild-souches", sep=""), quote = FALSE, sep = "\t", row.names = FALSE)

#pour filtrer directement les fréquences sous R
colnames(Data.filter)=c("S13", "S84", "S47", "S49", "S85", "SIT", "R47", "R53", "R44_1", "R44_2", "R44_3", "RIT")
#on filtre ce que l'on veut
List.V1<-subset(Data.filter, (R47>S13) & (R47>S84) & (R47>S47) & (R47>S49) & (R47>S85) & (R47>SIT) & (R47<R44_3))
```

#L'intersect se fait avec le BED contre un fichier GFF/GTF comme un transcriptome par exemple.

Très léger, ça passe en frontal:

module load bioinfo/bedtools-2.27.1
bedtools intersect -wb -a overlap.bed -b ~/save/Ref_Carpo/Cydia_pomonella-Ref.gff3 > overlap.intersect
#Même chose le GFF3 Gensas.
bedtools intersect -wb -a RY-3rep_overlap.bed -b Annotations_Gensas.gff3 > RY-3rep_Gensas.intersect


On obtient:
chr1    314013    314014    chr1    StringTie   transcript 314014
314014  1000    -   .   gene_id "MSTRG.34"; transcript_id "MSTRG.34.2";
chr1    576413    576414    chr1    StringTie   transcript 576414
576414  1000    +   .   gene_id "MSTRG.52"; transcript_id "MSTRG.52.1";
chr1    576413    576414    chr1    StringTie   transcript 576414
576414  1000    +   .   gene_id "MSTRG.53"; transcript_id "MSTRG.53.1";

# récupérer une liste des genes ID de la colonne 13 en gardant l'ordre initial (plus facile pour s'y retrouver)
#recupère la colonne 13 | enlèbve les " et ; | garde une seule valeur en cas de doublon (sans tri) > enregistre
awk '{print $13}' overlap.intersect | sed -e 's/"\|;//g' - | awk '!a[$0]++' -  > overlap.gene.list

# récupérer une liste des transcript ID de la colonne 15 en gardant l'ordre initial (plus facile pour s'y retrouver)
#recupère la colonne 15 | enlèbve les " et ; | garde une seule valeur en cas de doublon (sans tri) > enregistre
awk '{print $15}' overlap.intersect | sed -e 's/"\|;//g' - | awk '!a[$0]++' -  > overlap.transcript.list

# extraire les séquences correspondantes à cette liste depuis un fasta

module load bioinfo/seqtk-1.3
seqtk subseq ~/work/Pipe_RNAseq/Transcriptome_GTF/Merged/Cpom_Full_2021_transcripts.fasta cds.list > cds.fas
#Comparaison entre RUN
```{r}
seed2001.Cov1=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-2001-31274562/", "GWAS-13pops-chr1.Betai-Cov1.merged.sorted", sep=""),h=T)
seed2001.Cov2=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-2001-31274562/", "GWAS-13pops-chr1.Betai-Cov2.merged.sorted", sep=""),h=T)
seed5001.Cov1=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-5001-31250906/", "GWAS-13pops-chr1.Betai-Cov1.merged.sorted", sep=""),h=T)
seed5001.Cov2=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-5001-31250906/", "GWAS-13pops-chr1.Betai-Cov2.merged.sorted", sep=""),h=T)
seed8001.Cov1=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-8001-31430093/", "GWAS-13pops-chr1.Betai-Cov1.merged.sorted", sep=""),h=T)
seed8001.Cov2=read.table(paste("C:/BayPass_pipeline/Output/13pops/seed-8001-31430093/", "GWAS-13pops-chr1.Betai-Cov2.merged.sorted", sep=""),h=T)
Datmp13 = subset(seed2001.Cov1, BF.dB.>7.5)
Datmp14 = subset(seed2001.Cov1, BF.dB.>15)
Datmp15 = subset(seed2001.Cov2, BF.dB.>7.5)
Datmp16 = subset(seed2001.Cov2, BF.dB.>15)
Datmp17 = subset(seed5001.Cov1, BF.dB.>7.5)
Datmp18 = subset(seed5001.Cov1, BF.dB.>15)
Datmp19 = subset(seed5001.Cov2, BF.dB.>7.5)
Datmp20 = subset(seed5001.Cov2, BF.dB.>15)
Datmp21 = subset(seed8001.Cov1, BF.dB.>7.5)
Datmp22 = subset(seed8001.Cov1, BF.dB.>15)
Datmp23 = subset(seed8001.Cov2, BF.dB.>7.5)
Datmp24 = subset(seed8001.Cov2, BF.dB.>15)
Datmp25 = subset(C2_BF_Join1, M_C2>thresh.C2)
Datmp26 = subset(C2_BF_Join1, BF.dB.>15)
Datmp27 = subset(C2_BF_Join2, BF.dB.>15)
Datmp28 = subset(C2_BF_Join1, BF.dB.>7.5)
Datmp29 = subset(C2_BF_Join2, BF.dB.>7.5)


List.SNP13<-paste(Datmp13[,1], "_", Datmp13[,2], sep="") #2001-BF1 7.5
List.SNP14<-paste(Datmp14[,1], "_", Datmp14[,2], sep="") #2001-BF1 15
List.SNP15<-paste(Datmp13[,1], "_", Datmp15[,2], sep="") #2001-BF2 7.5
List.SNP16<-paste(Datmp14[,1], "_", Datmp16[,2], sep="") #2001-BF2 15
List.SNP17<-paste(Datmp17[,1], "_", Datmp17[,2], sep="") #5001-BF1 7.5
List.SNP18<-paste(Datmp18[,1], "_", Datmp18[,2], sep="") #5001-BF1 15
List.SNP19<-paste(Datmp19[,1], "_", Datmp19[,2], sep="") #5001-BF2 7.5
List.SNP20<-paste(Datmp20[,1], "_", Datmp20[,2], sep="") #5001-BF2 15
List.SNP21<-paste(Datmp21[,1], "_", Datmp21[,2], sep="") #8001-BF1 7.5
List.SNP22<-paste(Datmp22[,1], "_", Datmp22[,2], sep="") #8001-BF1 15
List.SNP23<-paste(Datmp23[,1], "_", Datmp23[,2], sep="") #8001-BF2 7.5
List.SNP24<-paste(Datmp24[,1], "_", Datmp24[,2], sep="") #8001-BF2 15
List.SNP25<-paste(Datmp25[,1], "_", Datmp25[,2], sep="") #C2 top5%
List.SNP26<-paste(Datmp26[,1], "_", Datmp26[,2], sep="") #BF1 > 15
List.SNP27<-paste(Datmp26[,1], "_", Datmp27[,2], sep="") #BF2 > 15
List.SNP28<-paste(Datmp28[,1], "_", Datmp28[,2], sep="") #BF1 > 7.5
List.SNP29<-paste(Datmp29[,1], "_", Datmp29[,2], sep="") #BF2 > 7.5
```

```{r venn}
#Diagramme de Venn
#cat.col = c("darkgreen", "black", "darkblue")
vd5 <- venn.diagram(x=list("2001-Cov1" = List.SNP14, "5001-Cov1" = List.SNP18, "8001-Cov1" = List.SNP22), fill = c("blue", "green", "orange"), cat.col = c("blue", "green", "orange"), cat.cex = 1.5, fontface = "bold", filename = NULL)
vd6 <- venn.diagram(x=list("2001-Cov2" = List.SNP16, "5001-Cov2" = List.SNP20, "8001-Cov2" = List.SNP24), fill = c("blue", "green", "orange"), cat.col = c("blue", "green", "orange"), cat.cex = 1.5, fontface = "bold", filename = NULL)
vd7 <- venn.diagram(x=list("2001-Cov2" = List.SNP15, "5001-Cov2" = List.SNP19, "8001-Cov2" = List.SNP23), fill = c("blue", "green", "orange"), cat.col = c("blue", "green", "orange"), cat.cex = 1.5, fontface = "bold", filename = NULL)
vd8 <- venn.diagram(x=list("2001-Cov1" = List.SNP13, "5001-Cov1" = List.SNP17, "8001-Cov1" = List.SNP21), fill = c("blue", "green", "orange"), cat.col = c("blue", "green", "orange"), cat.cex = 1.5, fontface = "bold", filename = NULL)
grid.newpage()
grid.draw(vd8)

```
#Création d'un jeu de données simulées "POD"
La fonction geno2YN convertit les données de comptages brutes en « Pseudo-Observed Data » (POD) et la fonction simulate.baypass génère un jeu de données simulées à partir de la matrice Ω déjà calculée (omegaB) ainsi qu'un constante Pi.beta que l'on récupère dans un des fichiers de sorties.
```{r POD data}
source("C:/BayPass_pipeline/utils/baypass_utils.R")
POD.data=geno2YN(paste0(path_input, "autosomes/genobaypass"))
pi.betaK=read.table(paste0(path_out, "autosomes/13pops-auto_STD_3cov.sub73_summary_beta_params.out"),h=T)$Mean
POD_BayPass<-simulate.baypass(omega.mat=omegaB,nsnp = 5000, beta.coef = NA, beta.pi = pi.betaK, sample.size=POD.data$NN, pi.maf=0, suffix="Poolseq.POD" )
```
Les 4 fichiers .POD générés (dans le dossier actif du pipeline) sont à copier sur le cluster de calcul. Le fichier G.Poolseq.POD sera analysé de la même manière que le jeu de données initial si ce n’est qu’il n’est pas nécessaire de découper ni de paralléliser l'analyse, le reste des paramètres (contraste, ecotype…) doit être identique.
#Analyse des résultats POD:
Les fichiers résultats POD_mat_omega, POD_summary_pi_xtx.out, sont copiés tel quel en local, le fichier POD_summary_contrast.out s’il contient les résultats de plusieurs contrastes doit être subdivisé par contraste (et renommés en ..POD_1..; ..POD2.. Etc…) au préalable pour obtenir une sortie par combinaison C2 qui seront aussi copiées en local.
```{r Thresholds}
source("C:/BayPass_pipeline/utils/baypass_utils.R")
POD.omega=as.matrix(read.table(paste(path_POD, "chr1/13pops_chr1.POD_mat_omega.out", sep="")))
plot(POD.omega,omegaB) ; abline(a=0,b=1)
FMD.POD <- fmd.dist(POD.omega,omegaB)
cat("Distance FMD =", FMD.POD, "\n")
#Seuil XtX top 1%
POD.XtX=read.table(paste(path_POD, "chr1/13pops_chr1.POD_summary_pi_xtx.out", sep=""),h=T)$M_XtX
thresh.XtX1=quantile(POD.XtX,probs=0.99)
cat("Seuil XtX =", thresh.XtX1, "(Max=", max(POD.XtX) ,")", "\n")
POD.XtX=read.table(paste(path_POD, "chr1/13pops_chr1.POD_summary_pi_xtx.out", sep=""),h=T)$M_XtX
thresh.XtX5=quantile(POD.XtX,probs=0.95)
cat("Seuil XtX =", thresh.XtX5, "(Max=", max(POD.XtX) ,")", "\n")

#Seuil C2 top 1%, pour chaque combinaison de contraste (3 dans l'exemple)
POD.C2=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_contrast.out"),h=T)$M_C2
thresh.C2_1=quantile(POD.C2,probs=0.99)
cat("Seuil C2 =", thresh.C2_1, "(Max=", max(POD.C2) ,")", "\n")
POD.C2=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_contrast.out"),h=T)$M_C2
thresh.C2_5=quantile(POD.C2,probs=0.95)
cat("Seuil C2 =", thresh.C2_5, "(Max=", max(POD.C2) ,")", "\n")

#Seuil BF top 1%, pour chaque combinaison de contraste (3 dans l'exemple)
POD.BF=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_betai_M"),h=T)$BF.dB.
thresh.BF1=quantile(POD.BF,probs=0.99)
cat("Seuil BF M =", thresh.BF1, "(Max=", max(POD.BF) ,")", "\n")
POD.BF=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_betai_M"),h=T)$BF.dB.
thresh.BF5=quantile(POD.BF,probs=0.95)
cat("Seuil BF M =", thresh.BF5, "(Max=", max(POD.BF) ,")", "\n")

POD.BF=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_betai_V15"),h=T)$BF.dB.
thresh.BF1=quantile(POD.BF,probs=0.99)
cat("Seuil BF V15 =", thresh.BF1, "(Max=", max(POD.BF) ,")", "\n")
POD.BF=read.table(paste0(path_POD, "chr1/13pops_chr1.POD_summary_betai_V15"),h=T)$BF.dB.
thresh.BF5=quantile(POD.BF,probs=0.95)
cat("Seuil BF V15 =", thresh.BF5, "(Max=", max(POD.BF) ,")", "\n")
```
La matrice Ω POD est comparée à la matrice Ω initialement calculée (omegaB) afin de valider la similarité des analyses. La fonction quantile calcule le seuil en fonction de la valeur probs qu’on lui donne : probs = 0,99 pour un seuil à 1%, probs=0,999 pour un seuil à 0,1% etc. Enfin on affiche un plot de la corrélation entre les deux matrice Ω, le seuil calculé pour la statistique XtX et pour toutes les combinaisons de contraste C2. Le calcul d'un seuil pour le les Bayes Factors n'est pas pertinent.
