<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://jolivares-inrae.github.io/BayPass_Tutorial/C:/Users/Olivares/Documents/R/Git_Work/BayPass_Tutorial/_book/images/icone3.png" />
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="https://github.com/Jolivares-INRAE/BayPass_Tutorial" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="twitter:image" content="https://jolivares-inrae.github.io/BayPass_Tutorial/C:/Users/Olivares/Documents/R/Git_Work/BayPass_Tutorial/_book/images/icone3.png" />

<meta name="author" content="Jérôme OLIVARES" />


<meta name="date" content="2023-07-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="données-brutes.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Résumé</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#mots-clefs"><i class="fa fa-check"></i>mots clefs</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#prérequis"><i class="fa fa-check"></i>prérequis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="présentation-du-logiciel-baypass-2.html"><a href="présentation-du-logiciel-baypass-2.html"><i class="fa fa-check"></i>Présentation du logiciel BayPass 2.3</a>
<ul>
<li class="chapter" data-level="" data-path="présentation-du-logiciel-baypass-2.html"><a href="présentation-du-logiciel-baypass-2.html#présentation-générale-de-lanalyse"><i class="fa fa-check"></i>Présentation générale de l’analyse :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html"><i class="fa fa-check"></i>Données brutes</a>
<ul>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html#obtention-dun-fichier-poolseq.vcf"><i class="fa fa-check"></i>Obtention d’un fichier Poolseq.vcf :</a></li>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html#filtrage-des-données-brutes"><i class="fa fa-check"></i>Filtrage des données brutes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html"><i class="fa fa-check"></i>Annexes</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#An1"><i class="fa fa-check"></i>Annexe 1</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#pairwise"><i class="fa fa-check"></i>Pairwise</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="annexes" class="section level1 unnumbered hasAnchor">
<h1>Annexes<a href="annexes.html#annexes" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="An1" class="section level2 unnumbered hasAnchor">
<h2>Annexe 1<a href="annexes.html#An1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A partir de cet objet pooldata on faire une première analyse des Fst.
Des outils sont décrits et exemplifiés dans la vignette de PoolFstat : (<a href="https://cran.r-project.org/web/packages/poolfstat/vignettes/vignette.pdf" class="uri">https://cran.r-project.org/web/packages/poolfstat/vignettes/vignette.pdf</a>), parmis ceux-ci on trouve l’analyse des Fst.</p>
<div id="pairwise" class="section level3 unnumbered hasAnchor">
<h3>Pairwise<a href="annexes.html#pairwise" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Calcul des Fst entre les populations deux à deux (pairwise) afin de visualiser la proximité génétique entre populations.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="annexes.html#cb8-1" tabindex="-1"></a><span class="co">#Calcul des pairwise Fst </span></span>
<span id="cb8-2"><a href="annexes.html#cb8-2" tabindex="-1"></a>PairWise.fst <span class="ot">&lt;-</span> <span class="fu">compute.pairwiseFST</span>(GWAS.pooldata, <span class="at">method =</span> <span class="st">&quot;Anova&quot;</span>, <span class="at">min.cov.per.pool =</span> <span class="dv">4</span>, <span class="at">max.cov.per.pool =</span> <span class="fl">1e+06</span>, <span class="at">min.maf =</span> <span class="fl">0.05</span>, <span class="at">output.snp.values =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Computation of the 78 pairwise Fst
## 
## Overall Analysis Time: 0 h 0 m 0 s</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="annexes.html#cb10-1" tabindex="-1"></a><span class="co">#conversion en matrice de distance</span></span>
<span id="cb10-2"><a href="annexes.html#cb10-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">dist</span>(<span class="fu">t</span>(PairWise.fst<span class="sc">@</span>PairwiseFSTmatrix)))</span>
<span id="cb10-3"><a href="annexes.html#cb10-3" tabindex="-1"></a>df</span></code></pre></div>
<pre><code>##            pop01      pop02      pop03      pop04      pop05      pop06
## pop01 0.00000000 0.02599730 0.04047179 0.03663751 0.04296871 0.03467895
## pop02 0.02599730 0.00000000 0.02163574 0.01668035 0.03006506 0.05535359
## pop03 0.04047179 0.02163574 0.00000000 0.00972097 0.04251282 0.06408399
## pop04 0.03663751 0.01668035 0.00972097 0.00000000 0.03987384 0.05859326
## pop05 0.04296871 0.03006506 0.04251282 0.03987384 0.00000000 0.06971754
## pop06 0.03467895 0.05535359 0.06408399 0.05859326 0.06971754 0.00000000
## pop07 0.03053708 0.01020988 0.01539477 0.01103366 0.03480062 0.05686072
## pop08 0.04351450 0.04143124 0.03574270 0.03355561 0.06865341 0.05031445
## pop09 0.01568180 0.03339984 0.04601002 0.04203211 0.04237655 0.03585086
## pop10 0.04294600 0.03300989 0.04295159 0.04308963 0.01779158 0.06501738
## pop11 0.04198359 0.03953978 0.05263064 0.04737610 0.04070589 0.05618353
## pop12 0.04274732 0.04230794 0.05201989 0.04877386 0.03934474 0.05856191
## pop13 0.02928271 0.03766647 0.05230044 0.04811739 0.05368314 0.02395252
##            pop07      pop08      pop09      pop10       pop11       pop12
## pop01 0.03053708 0.04351450 0.01568180 0.04294600 0.041983592 0.042747321
## pop02 0.01020988 0.04143124 0.03339984 0.03300989 0.039539781 0.042307937
## pop03 0.01539477 0.03574270 0.04601002 0.04295159 0.052630643 0.052019892
## pop04 0.01103366 0.03355561 0.04203211 0.04308963 0.047376104 0.048773856
## pop05 0.03480062 0.06865341 0.04237655 0.01779158 0.040705889 0.039344737
## pop06 0.05686072 0.05031445 0.03585086 0.06501738 0.056183526 0.058561908
## pop07 0.00000000 0.03819477 0.03736103 0.03891893 0.044734171 0.046729487
## pop08 0.03819477 0.00000000 0.04864457 0.06510535 0.057654706 0.060230565
## pop09 0.03736103 0.04864457 0.00000000 0.03904249 0.039467063 0.040213648
## pop10 0.03891893 0.06510535 0.03904249 0.00000000 0.025915893 0.023573077
## pop11 0.04473417 0.05765471 0.03946706 0.02591589 0.000000000 0.007042681
## pop12 0.04672949 0.06023056 0.04021365 0.02357308 0.007042681 0.000000000
## pop13 0.04261189 0.05247061 0.02609580 0.05261074 0.051519712 0.050635147
##            pop13
## pop01 0.02928271
## pop02 0.03766647
## pop03 0.05230044
## pop04 0.04811739
## pop05 0.05368314
## pop06 0.02395252
## pop07 0.04261189
## pop08 0.05247061
## pop09 0.02609580
## pop10 0.05261074
## pop11 0.05151971
## pop12 0.05063515
## pop13 0.00000000</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="annexes.html#cb12-1" tabindex="-1"></a><span class="co">#heatmap</span></span>
<span id="cb12-2"><a href="annexes.html#cb12-2" tabindex="-1"></a>cim_color <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Reds&quot;</span>)))(<span class="dv">9</span>)</span>
<span id="cb12-3"><a href="annexes.html#cb12-3" tabindex="-1"></a><span class="fu">cim</span>(df, <span class="at">color =</span> cim_color, <span class="at">symkey =</span> <span class="cn">FALSE</span>, <span class="at">margins =</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">10</span>),  <span class="at">title =</span> <span class="st">&quot;heatmap des pairwise Fst entre les populations&quot;</span>)</span></code></pre></div>
<p><img src="BayPass_Fr_files/figure-html/heatmap-PW-fst-1.png" width="960" /></p>
<p>On peut aussi calculer et plotter des Fst multi-locus en balayant le génome avec une fenêtre glissante de SNP consécutifs, une région génomique très différenciée apparaitra sous la forme d’une éruption de points colorés.
## Fst en fenêtre glissante {-}</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="annexes.html#cb13-1" tabindex="-1"></a><span class="co">#calcul des Fst avec une fenêtre glissante de 100 SNP</span></span>
<span id="cb13-2"><a href="annexes.html#cb13-2" tabindex="-1"></a>Multi.Loc.fst <span class="ot">&lt;-</span> <span class="fu">computeFST</span>(GWAS.pooldata, <span class="at">method =</span> <span class="st">&quot;Anova&quot;</span>, <span class="at">sliding.window.size =</span> <span class="dv">100</span>)</span></code></pre></div>
<pre><code>## Start sliding-window scan
## 1 chromosomes scanned (with more than 100 SNPs)
## 
## Average (min-max) Window Sizes 28 ( 1.7 - 865.6 ) kb</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="annexes.html#cb15-1" tabindex="-1"></a><span class="co">#conversion en objet data frame</span></span>
<span id="cb15-2"><a href="annexes.html#cb15-2" tabindex="-1"></a>df.fst<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(Multi.Loc.fst<span class="sc">$</span>sliding.windows.fst, <span class="at">h=</span>T)</span>
<span id="cb15-3"><a href="annexes.html#cb15-3" tabindex="-1"></a><span class="co">#plot en ligne.</span></span>
<span id="cb15-4"><a href="annexes.html#cb15-4" tabindex="-1"></a>Fst.plot <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>df.fst, <span class="fu">aes</span>(<span class="at">x=</span>CumulatedPosition<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">y=</span>MultiLocusFst)) <span class="sc">+</span></span>
<span id="cb15-5"><a href="annexes.html#cb15-5" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Chr), <span class="at">alpha=</span><span class="fl">0.8</span>, <span class="at">size=</span><span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb15-6"><a href="annexes.html#cb15-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Fst en fenêtres glissantes&quot;</span>) <span class="sc">+</span></span>
<span id="cb15-7"><a href="annexes.html#cb15-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span>Multi.Loc.fst<span class="sc">$</span>FST,<span class="at">lty=</span><span class="dv">2</span>) <span class="co">#le seuil indique la Fst globale estimée à l&#39;échelle du génome</span></span>
<span id="cb15-8"><a href="annexes.html#cb15-8" tabindex="-1"></a>Fst.plot <span class="sc">+</span> <span class="fu">scale_x_continuous</span>() <span class="sc">+</span> <span class="fu">scale_y_continuous</span>() <span class="sc">+</span></span>
<span id="cb15-9"><a href="annexes.html#cb15-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb15-10"><a href="annexes.html#cb15-10" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Chr, <span class="at">scales =</span> <span class="st">&#39;free_x&#39;</span>, <span class="at">strip.position =</span><span class="fu">c</span>(<span class="st">&quot;bottom&quot;</span>)) <span class="co">#affichage des chromosomes en grille</span></span></code></pre></div>
<p><img src="BayPass_Fr_files/figure-html/plot-sliding-Fst-1.png" width="672" /></p>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Futschik, Andreas, and Christian Schlötterer. 2010. <span>“The <span>Next</span> <span>Generation</span> of <span>Molecular</span> <span>Markers</span> <span>From</span> <span>Massively</span> <span>Parallel</span> <span>Sequencing</span> of <span>Pooled</span> <span>DNA</span> <span>Samples</span>.”</span> <em>Genetics</em> 186 (1): 207–18. <a href="https://doi.org/10.1534/genetics.110.114397">https://doi.org/10.1534/genetics.110.114397</a>.
</div>
<div class="csl-entry">
Gautier, Mathieu. 2015. <span>“Genome-<span>Wide</span> <span>Scan</span> for <span>Adaptive</span> <span>Divergence</span> and <span>Association</span> with <span>Population</span>-<span>Specific</span> <span>Covariates</span>.”</span> <em>Genetics</em> 201 (4): 1555–79. <a href="https://doi.org/10.1534/genetics.115.181453">https://doi.org/10.1534/genetics.115.181453</a>.
</div>
<div class="csl-entry">
Gautier, Mathieu, Renaud Vitalis, Laurence Flori, and Arnaud Estoup. 2022. <span>“F-Statistics Estimation and Admixture Graph Construction with <span>Pool</span>-<span>Seq</span> or Allele Count Data Using the <span>R</span> Package Poolfstat.”</span>
</div>
<div class="csl-entry">
Koboldt, Daniel C., Qunyuan Zhang, David E. Larson, Dong Shen, Michael D. McLellan, Ling Lin, Christopher A. Miller, Elaine R. Mardis, Li Ding, and Richard K. Wilson. 2012. <span>“<span>VarScan</span> 2: <span>Somatic</span> Mutation and Copy Number Alteration Discovery in Cancer by Exome Sequencing.”</span> <em>Genome Research</em> 22 (3): 568–76. <a href="https://doi.org/10.1101/gr.129684.111">https://doi.org/10.1101/gr.129684.111</a>.
</div>
<div class="csl-entry">
Li, Heng, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, and 1000 Genome Project Data Processing Subgroup. 2009. <span>“The <span>Sequence</span> <span>Alignment</span>/<span>Map</span> Format and <span>SAMtools</span>.”</span> <em>Bioinformatics</em> 25 (16): 2078–79. <a href="https://doi.org/10.1093/bioinformatics/btp352">https://doi.org/10.1093/bioinformatics/btp352</a>.
</div>
<div class="csl-entry">
Nielsen, Erica S. 2020. <span>“Pool-<span>Seq</span> <span>Analyses</span>: <span>PoolFstat</span> &amp; <span>BayPass</span>.”</span> <a href="https://esnielsen.github.io/post/pool-seq-analyses-poolfstat-baypass/">https://esnielsen.github.io/post/pool-seq-analyses-poolfstat-baypass/</a>.
</div>
</div>
</div>
</div>
</div>






            </section>

          </div>
        </div>
      </div>
<a href="données-brutes.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/10-Annexes.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BayPass_Fr.pdf", "BayPass_Fr.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
