<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.35 and GitBook 2.6.7" />

  <meta property="og:title" content="Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://jolivares-inrae.github.io/BayPass_Tutorial/C:/Users/Olivares/Documents/R/Git_Work/BayPass_Tutorial/cover.jpg" />
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="https://github.com/Jolivares-INRAE/BayPass_Tutorial" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Annexes | BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="twitter:image" content="https://jolivares-inrae.github.io/BayPass_Tutorial/C:/Users/Olivares/Documents/R/Git_Work/BayPass_Tutorial/cover.jpg" />

<meta name="author" content="Jérôme OLIVARES" />


<meta name="date" content="2023-10-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="approche-par-fenêtres-glissantes.html"/>
<link rel="next" href="références.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
\usepackage{booktabs}


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Résumé</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#mots-clefs"><i class="fa fa-check"></i>mots clefs</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#prérequis"><i class="fa fa-check"></i>prérequis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="présentation-du-logiciel-baypass-2.html"><a href="présentation-du-logiciel-baypass-2.html"><i class="fa fa-check"></i>Présentation du logiciel BayPass 2.3</a>
<ul>
<li class="chapter" data-level="" data-path="présentation-du-logiciel-baypass-2.html"><a href="présentation-du-logiciel-baypass-2.html#présentation-générale-de-lanalyse"><i class="fa fa-check"></i>Présentation générale de l’analyse :</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html"><i class="fa fa-check"></i>Données brutes</a>
<ul>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html#obtention-dun-fichier-poolseq.vcf"><i class="fa fa-check"></i>Obtention d’un fichier Poolseq.vcf :</a></li>
<li class="chapter" data-level="" data-path="données-brutes.html"><a href="données-brutes.html#filtrage-des-données-brutes"><i class="fa fa-check"></i>Filtrage des données brutes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html"><a href="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html"><i class="fa fa-check"></i>Conversion du pooldata en fichiers d’entrées pour BayPass</a>
<ul>
<li class="chapter" data-level="" data-path="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html"><a href="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html#design-de-lanalyse-baypass"><i class="fa fa-check"></i>Design de l’analyse BayPass</a></li>
<li class="chapter" data-level="" data-path="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html"><a href="conversion-du-pooldata-en-fichiers-dentrées-pour-baypass.html#subsampling"><i class="fa fa-check"></i>Subsampling</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="baypass-lanalyse-poolseq.html"><a href="baypass-lanalyse-poolseq.html"><i class="fa fa-check"></i>Baypass : l’analyse poolseq</a></li>
<li class="chapter" data-level="" data-path="baypass-les-résultats.html"><a href="baypass-les-résultats.html"><i class="fa fa-check"></i>Baypass : les résultats</a>
<ul>
<li class="chapter" data-level="" data-path="baypass-les-résultats.html"><a href="baypass-les-résultats.html#validation-des-données-analysées"><i class="fa fa-check"></i>Validation des données analysées</a></li>
<li class="chapter" data-level="" data-path="baypass-les-résultats.html"><a href="baypass-les-résultats.html#concaténation-des-résultats"><i class="fa fa-check"></i>Concaténation des résultats</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="exploitation-des-résultats.html"><a href="exploitation-des-résultats.html"><i class="fa fa-check"></i>Exploitation des résultats</a>
<ul>
<li class="chapter" data-level="" data-path="exploitation-des-résultats.html"><a href="exploitation-des-résultats.html#les-résultats-de-différentiation-xtx"><i class="fa fa-check"></i>Les résultats de différentiation <em>XtX</em>:</a></li>
<li class="chapter" data-level="" data-path="exploitation-des-résultats.html"><a href="exploitation-des-résultats.html#les-résultats-de-bayes-factor-bf"><i class="fa fa-check"></i>Les résultats de Bayes Factor <em>BF</em>:</a></li>
<li class="chapter" data-level="" data-path="exploitation-des-résultats.html"><a href="exploitation-des-résultats.html#les-résultats-de-contraste-c2"><i class="fa fa-check"></i>Les résultats de Contraste <em>C2</em>:</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="approche-par-fenêtres-glissantes.html"><a href="approche-par-fenêtres-glissantes.html"><i class="fa fa-check"></i>Approche par fenêtres glissantes</a></li>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html"><i class="fa fa-check"></i>Annexes</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#An1"><i class="fa fa-check"></i>Annexe 1</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#fst-en-fenêtre-glissante"><i class="fa fa-check"></i>Fst en fenêtre glissante</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#An2"><i class="fa fa-check"></i>Annexe 2</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#conversion-de-la-matrice-svd-en-graphique-ggplot"><i class="fa fa-check"></i>Conversion de la matrice SVD en graphique ggplot</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#An3"><i class="fa fa-check"></i>Annexe 3</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#concaténages-des-résultats-betai-et-contrastes"><i class="fa fa-check"></i>Concaténages des résultats Betai et contrastes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#An4"><i class="fa fa-check"></i>Annexe 4</a>
<ul>
<li class="chapter" data-level="" data-path="annexes.html"><a href="annexes.html#création-et-analyse-dun-jeu-de-données-pseudo-observées-pod"><i class="fa fa-check"></i>Création et analyse d’un jeu de données pseudo-observées (POD)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="références.html"><a href="références.html"><i class="fa fa-check"></i>Références</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">BayPass 2.3: tutoriel de génomique d’association adapté au séquençage en pool</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="annexes" class="section level1 unnumbered hasAnchor">
<h1>Annexes<a href="annexes.html#annexes" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="An1" class="section level2 unnumbered hasAnchor">
<h2>Annexe 1<a href="annexes.html#An1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Il est possible de calculer des Fst multi-locus en balayant le génome avec une fenêtre glissante de SNP consécutifs et un chevauchement d’une demi fenêtre, une fois ploter, une région génomique très différenciée apparaîtra sous la forme d’une éruption de points colorés.</p>
<div id="fst-en-fenêtre-glissante" class="section level3 unnumbered hasAnchor">
<h3>Fst en fenêtre glissante<a href="annexes.html#fst-en-fenêtre-glissante" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb39"><pre class="sourceCode r bg-chunk"><code class="sourceCode r"><span id="cb39-1"><a href="annexes.html#cb39-1" tabindex="-1"></a><span class="co">#calcul des Fst avec une fenêtre glissante de 100 SNP consécutifs .</span></span>
<span id="cb39-2"><a href="annexes.html#cb39-2" tabindex="-1"></a>Multi.Loc.fst <span class="ot">&lt;-</span> <span class="fu">computeFST</span>(pooldata,</span>
<span id="cb39-3"><a href="annexes.html#cb39-3" tabindex="-1"></a>                            <span class="at">method =</span> <span class="st">&quot;Anova&quot;</span>,</span>
<span id="cb39-4"><a href="annexes.html#cb39-4" tabindex="-1"></a>                            <span class="at">sliding.window.size =</span> <span class="dv">100</span>)</span></code></pre></div>
<pre class="bg-output"><code>## Start sliding-window scan
## 4 chromosomes scanned (with more than 100 SNPs)
## 
## Average (min-max) Window Sizes 9.3 ( 1.8 - 2751 ) kb</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r bg-chunk"><code class="sourceCode r"><span id="cb41-1"><a href="annexes.html#cb41-1" tabindex="-1"></a><span class="co">#conversion en objet data frame</span></span>
<span id="cb41-2"><a href="annexes.html#cb41-2" tabindex="-1"></a>df.fst<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(Multi.Loc.fst<span class="sc">$</span>sliding.windows.fst, <span class="at">h=</span>T)</span>
<span id="cb41-3"><a href="annexes.html#cb41-3" tabindex="-1"></a><span class="co">#plot.</span></span>
<span id="cb41-4"><a href="annexes.html#cb41-4" tabindex="-1"></a>Fst.plot <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>df.fst, <span class="fu">aes</span>(<span class="at">x=</span>CumulatedPosition<span class="sc">/</span><span class="fl">1e6</span>, <span class="at">y=</span>MultiLocusFst)) <span class="sc">+</span></span>
<span id="cb41-5"><a href="annexes.html#cb41-5" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Chr), <span class="at">alpha=</span><span class="fl">0.8</span>, <span class="at">size=</span><span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb41-6"><a href="annexes.html#cb41-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Fst en fenêtres glissantes&quot;</span>) <span class="sc">+</span></span>
<span id="cb41-7"><a href="annexes.html#cb41-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span>Multi.Loc.fst<span class="sc">$</span>FST,<span class="at">lty=</span><span class="dv">2</span>) <span class="co">#le seuil indique la Fst globale estimée à l&#39;échelle du génome</span></span>
<span id="cb41-8"><a href="annexes.html#cb41-8" tabindex="-1"></a>Fst.plot <span class="sc">+</span> <span class="fu">scale_x_continuous</span>() <span class="sc">+</span></span>
<span id="cb41-9"><a href="annexes.html#cb41-9" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>() <span class="sc">+</span></span>
<span id="cb41-10"><a href="annexes.html#cb41-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(), <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb41-11"><a href="annexes.html#cb41-11" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Chr, <span class="at">scales =</span> <span class="st">&#39;free_x&#39;</span>, <span class="at">strip.position =</span><span class="fu">c</span>(<span class="st">&quot;bottom&quot;</span>)) <span class="co">#affichage des chromosomes en grille</span></span></code></pre></div>
<p><img src="BayPass_Fr_files/figure-html/plot-sliding-Fst-1.png" width="672" /></p>
</div>
</div>
<div id="An2" class="section level2 unnumbered hasAnchor">
<h2>Annexe 2<a href="annexes.html#An2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="conversion-de-la-matrice-svd-en-graphique-ggplot" class="section level3 unnumbered hasAnchor">
<h3>Conversion de la matrice SVD en graphique ggplot<a href="annexes.html#conversion-de-la-matrice-svd-en-graphique-ggplot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Il peut être intéressant de convertir la SVD en graphique ggplot, afin d’avoir la main sur tous les aspects cosmétiques en vue d’une publication par exemple.
Le principe consiste à créer une table dans laquelle on intègre les informations à ploter (noms, phenotype, couleurs) ainsi que les les 2 premiers vecteurs propres (eigen vector) et leur variance, de la matrice SVD.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="annexes.html#cb42-1" tabindex="-1"></a><span class="co"># convertit en data.frame</span></span>
<span id="cb42-2"><a href="annexes.html#cb42-2" tabindex="-1"></a>tab_SVD <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample.id =</span> pnames,</span>
<span id="cb42-3"><a href="annexes.html#cb42-3" tabindex="-1"></a>    <span class="co">#défini le phenotype de chaque population</span></span>
<span id="cb42-4"><a href="annexes.html#cb42-4" tabindex="-1"></a>    <span class="at">phenotype =</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;Sensible&quot;</span>,   <span class="st">&quot;Sensible&quot;</span>,  <span class="st">&quot;Sensible&quot;</span>, <span class="st">&quot;Resistant&quot;</span>, <span class="st">&quot;Resistant&quot;</span>, <span class="st">&quot;Resistant&quot;</span>)),</span>
<span id="cb42-5"><a href="annexes.html#cb42-5" tabindex="-1"></a>    <span class="co">#attribut une couleur à chaque zone géographique</span></span>
<span id="cb42-6"><a href="annexes.html#cb42-6" tabindex="-1"></a>    <span class="at">col_geo =</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&quot;orange&quot;</span>,   <span class="st">&quot;darkgreen&quot;</span>,  <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;orange&quot;</span>, <span class="st">&quot;darkgreen&quot;</span>,<span class="st">&quot;blue&quot;</span>)),</span>
<span id="cb42-7"><a href="annexes.html#cb42-7" tabindex="-1"></a>    <span class="at">PC =</span> SVD_omega<span class="sc">$</span>PC,    <span class="co"># the first eigenvector</span></span>
<span id="cb42-8"><a href="annexes.html#cb42-8" tabindex="-1"></a>    <span class="at">eig =</span> SVD_omega<span class="sc">$</span>eig,    <span class="co"># the second eigenvector</span></span>
<span id="cb42-9"><a href="annexes.html#cb42-9" tabindex="-1"></a>    <span class="at">VAR =</span> SVD_omega<span class="sc">$</span>pcent.var,    <span class="co"># variance de chaque eigenvector</span></span>
<span id="cb42-10"><a href="annexes.html#cb42-10" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb42-11"><a href="annexes.html#cb42-11" tabindex="-1"></a></span>
<span id="cb42-12"><a href="annexes.html#cb42-12" tabindex="-1"></a>ggplot_svd <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>tab_SVD, <span class="fu">aes</span>(<span class="at">x=</span>PC<span class="fl">.1</span>, <span class="at">y=</span>PC<span class="fl">.2</span>, <span class="at">shape=</span>phenotype, <span class="at">color=</span> col_geo)) <span class="sc">+</span></span>
<span id="cb42-13"><a href="annexes.html#cb42-13" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb42-14"><a href="annexes.html#cb42-14" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&#39;Sensible&#39;</span><span class="ot">=</span><span class="dv">79</span>, <span class="st">&#39;Resistant&#39;</span><span class="ot">=</span><span class="dv">16</span>))<span class="sc">+</span></span>
<span id="cb42-15"><a href="annexes.html#cb42-15" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">name =</span> <span class="st">&quot;region&quot;</span>,</span>
<span id="cb42-16"><a href="annexes.html#cb42-16" tabindex="-1"></a>                      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;sud&quot;</span>, <span class="st">&quot;nord&quot;</span>, <span class="st">&quot;est&quot;</span>),</span>
<span id="cb42-17"><a href="annexes.html#cb42-17" tabindex="-1"></a>                      <span class="at">values =</span> col_geo)<span class="sc">+</span></span>
<span id="cb42-18"><a href="annexes.html#cb42-18" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;Singular Value Decomposition of the covariance (&quot;</span>,<span class="fu">expression</span>(<span class="st">&quot;\U03A9&quot;</span>),<span class="st">&quot;) matrix&quot;</span>))</span>
<span id="cb42-19"><a href="annexes.html#cb42-19" tabindex="-1"></a></span>
<span id="cb42-20"><a href="annexes.html#cb42-20" tabindex="-1"></a>ggplot_svd <span class="sc">+</span></span>
<span id="cb42-21"><a href="annexes.html#cb42-21" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>()<span class="sc">+</span></span>
<span id="cb42-22"><a href="annexes.html#cb42-22" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>()<span class="sc">+</span></span>
<span id="cb42-23"><a href="annexes.html#cb42-23" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste</span>(<span class="st">&quot;PC1 (&quot;</span>,<span class="fu">round</span>(tab_SVD<span class="sc">$</span>VAR[<span class="dv">1</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span> , <span class="at">sep=</span><span class="st">&quot;&quot;</span>))<span class="sc">+</span></span>
<span id="cb42-24"><a href="annexes.html#cb42-24" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(tab_SVD<span class="sc">$</span>VAR[<span class="dv">2</span>], <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span> , <span class="at">sep=</span><span class="st">&quot;&quot;</span>))<span class="sc">+</span></span>
<span id="cb42-25"><a href="annexes.html#cb42-25" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> sample.id), <span class="at">nudge_y =</span> <span class="fl">0.05</span>,<span class="at">fontface =</span> <span class="st">&#39;bold&#39;</span>)</span></code></pre></div>
</div>
</div>
<div id="An3" class="section level2 unnumbered hasAnchor">
<h2>Annexe 3<a href="annexes.html#An3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="concaténages-des-résultats-betai-et-contrastes" class="section level3 unnumbered hasAnchor">
<h3>Concaténages des résultats Betai et contrastes<a href="annexes.html#concaténages-des-résultats-betai-et-contrastes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Valable pour le modèle standard pour Betai et pour les 3 modèles pour contraste.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="annexes.html#cb43-1" tabindex="-1"></a></span>
<span id="cb43-2"><a href="annexes.html#cb43-2" tabindex="-1"></a><span class="va">Snpdet_path</span><span class="op">=</span><span class="st">&#39;/../BayPass/Input/&#39;</span></span>
<span id="cb43-3"><a href="annexes.html#cb43-3" tabindex="-1"></a><span class="va">My_prefix</span><span class="op">=</span><span class="st">&#39;project_STD.sub&#39;</span></span>
<span id="cb43-4"><a href="annexes.html#cb43-4" tabindex="-1"></a></span>
<span id="cb43-5"><a href="annexes.html#cb43-5" tabindex="-1"></a><span class="co">#on boucle sur toutes les covariables ou combinaisons de contraste</span></span>
<span id="cb43-6"><a href="annexes.html#cb43-6" tabindex="-1"></a><span class="cf">for</span> k <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">3</span><span class="dt">}</span></span>
<span id="cb43-7"><a href="annexes.html#cb43-7" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb43-8"><a href="annexes.html#cb43-8" tabindex="-1"></a><span class="co">#on boucle sur tous les fichiers</span></span>
<span id="cb43-9"><a href="annexes.html#cb43-9" tabindex="-1"></a><span class="cf">for</span> i <span class="kw">in</span> <span class="dt">{</span><span class="dv">1</span><span class="dt">..</span><span class="dv">5</span><span class="dt">}</span></span>
<span id="cb43-10"><a href="annexes.html#cb43-10" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb43-11"><a href="annexes.html#cb43-11" tabindex="-1"></a><span class="va">file1</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$Snpdet_path</span><span class="st">/snpdet.sub&quot;&quot;</span><span class="va">$i</span><span class="st">&quot;&quot;&quot;</span><span class="va">)</span></span>
<span id="cb43-12"><a href="annexes.html#cb43-12" tabindex="-1"></a><span class="va">file2</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="st">&quot;</span><span class="va">$My_prefix</span><span class="st">&quot;&quot;</span><span class="va">$i</span><span class="st">&quot;&quot;_summary_betai_reg.out&quot;</span><span class="va">)</span>                          <span class="co">#adapter le préfixe</span></span>
<span id="cb43-13"><a href="annexes.html#cb43-13" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-e</span> <span class="st">&quot;s/[[:space:]]\+/ /g&quot;</span> <span class="va">$file1</span> <span class="op">&gt;</span> SubSNP.b                                  <span class="co">#on remplace tous les séparateurs par tabulation, ça évite les bugs</span></span>
<span id="cb43-14"><a href="annexes.html#cb43-14" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-e</span> <span class="st">&quot;s/[[:space:]]\+/ /g&quot;</span> <span class="va">$file2</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-vK</span><span class="op">=</span><span class="st">&quot;</span><span class="va">$k</span><span class="st">&quot;</span> <span class="st">&#39;{if($1 == K) {print}}&#39;</span> <span class="at">-</span> <span class="op">&gt;</span> SubData.b                <span class="co">#on prend si col 1 (covariable) = k</span></span>
<span id="cb43-15"><a href="annexes.html#cb43-15" tabindex="-1"></a><span class="fu">paste</span> SubSNP.b SubData.b <span class="op">&gt;&gt;</span> tmp-Cov<span class="va">$k</span>.merged                            <span class="co">#on colle/incrémente dans un nouveau fichier</span></span>
<span id="cb43-16"><a href="annexes.html#cb43-16" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb43-17"><a href="annexes.html#cb43-17" tabindex="-1"></a><span class="co">#on tri par pos et on reconstruit l&#39;entête</span></span>
<span id="cb43-18"><a href="annexes.html#cb43-18" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;{for (i=1;i&lt;=NF;i++) if ($i+0 == $i &amp;&amp; $i ~ /e/) $2 = sprintf(&quot;%.0f&quot;, $i)} 1&#39;</span> tmp-Cov<span class="va">$k</span>.merged <span class="kw">|</span> <span class="fu">sort</span> <span class="at">-k1,1</span> <span class="at">-k2,2n</span> <span class="at">-</span> <span class="op">&gt;</span> project_Betai.Cov<span class="va">$k</span>     </span>
<span id="cb43-19"><a href="annexes.html#cb43-19" tabindex="-1"></a></span>
<span id="cb43-20"><a href="annexes.html#cb43-20" tabindex="-1"></a><span class="co">#on vire les fichiers tmp</span></span>
<span id="cb43-21"><a href="annexes.html#cb43-21" tabindex="-1"></a><span class="fu">rm</span> tmp-Cov<span class="va">$k</span>.merged</span>
<span id="cb43-22"><a href="annexes.html#cb43-22" tabindex="-1"></a><span class="fu">rm</span> SubSNP.b</span>
<span id="cb43-23"><a href="annexes.html#cb43-23" tabindex="-1"></a><span class="fu">rm</span> SubData.b</span>
<span id="cb43-24"><a href="annexes.html#cb43-24" tabindex="-1"></a></span>
<span id="cb43-25"><a href="annexes.html#cb43-25" tabindex="-1"></a><span class="co">#entête pour les résultats Betai:</span></span>
<span id="cb43-26"><a href="annexes.html#cb43-26" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;1i chr pos All1 All2 COVARIABLE MRK M_Pearson SD_Pearson M_Spearman SD_Spearman BF(dB) Beta_is SD_Beta_is eBPis&#39;</span> project_Betai.Cov<span class="va">$k</span> </span>
<span id="cb43-27"><a href="annexes.html#cb43-27" tabindex="-1"></a></span>
<span id="cb43-28"><a href="annexes.html#cb43-28" tabindex="-1"></a><span class="co">#entête pour les résultats de contraste:</span></span>
<span id="cb43-29"><a href="annexes.html#cb43-29" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;1i chr pos All1 All2 CONTRAST MRK M_C2 SD_C2 C2_std C2_log10(1/pval)&#39;</span> project_contrast-C<span class="va">$k</span></span></code></pre></div>
</div>
</div>
<div id="An4" class="section level2 unnumbered hasAnchor">
<h2>Annexe 4<a href="annexes.html#An4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="création-et-analyse-dun-jeu-de-données-pseudo-observées-pod" class="section level3 unnumbered hasAnchor">
<h3>Création et analyse d’un jeu de données pseudo-observées (POD)<a href="annexes.html#création-et-analyse-dun-jeu-de-données-pseudo-observées-pod" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>La fonction geno2YN extrait les données de comptages brutes en « Pseudo-Observed Data » (POD) et la fonction simulate.baypass génère un jeu de données simulées à partir de la matrice Ω déjà calculée (omegaB) ainsi qu’un constante Pi.beta que l’on récupère dans un des fichiers de sorties.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="annexes.html#cb44-1" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;/../baypass_utils.R&quot;</span>)</span>
<span id="cb44-2"><a href="annexes.html#cb44-2" tabindex="-1"></a><span class="co">#extrait les données de comptage</span></span>
<span id="cb44-3"><a href="annexes.html#cb44-3" tabindex="-1"></a>POD.data<span class="ot">=</span><span class="fu">geno2YN</span>(<span class="fu">paste</span>(path_input, <span class="st">&quot;genobaypass&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>))</span>
<span id="cb44-4"><a href="annexes.html#cb44-4" tabindex="-1"></a><span class="co">#Extrait le beta pi moyen</span></span>
<span id="cb44-5"><a href="annexes.html#cb44-5" tabindex="-1"></a>pi.betaK<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(path_out, <span class="st">&quot;project.sub3_summary_beta_params.out&quot;</span>),<span class="at">h=</span>T)<span class="sc">$</span>Mean</span>
<span id="cb44-6"><a href="annexes.html#cb44-6" tabindex="-1"></a><span class="co">#créé un jeu de données de 10 000 SNPs</span></span>
<span id="cb44-7"><a href="annexes.html#cb44-7" tabindex="-1"></a>POD_BayPass<span class="ot">&lt;-</span><span class="fu">simulate.baypass</span>(<span class="at">omega.mat=</span>omegaB,</span>
<span id="cb44-8"><a href="annexes.html#cb44-8" tabindex="-1"></a>                              <span class="at">nsnp =</span> <span class="dv">10000</span>,</span>
<span id="cb44-9"><a href="annexes.html#cb44-9" tabindex="-1"></a>                              <span class="at">beta.coef =</span> <span class="cn">NA</span>,</span>
<span id="cb44-10"><a href="annexes.html#cb44-10" tabindex="-1"></a>                              <span class="at">beta.pi =</span> pi.betaK,</span>
<span id="cb44-11"><a href="annexes.html#cb44-11" tabindex="-1"></a>                              <span class="at">sample.size=</span>POD.data<span class="sc">$</span>NN,</span>
<span id="cb44-12"><a href="annexes.html#cb44-12" tabindex="-1"></a>                              <span class="at">pi.maf=</span><span class="dv">0</span>, <span class="at">suffix=</span><span class="st">&quot;project.POD&quot;</span> )</span></code></pre></div>
<p>Les 4 fichiers .POD générés (dans le dossier actif du pipeline) sont à copier sur le cluster de calcul. Le fichier G.project.POD sera analysé de la même manière que le jeu de données initial (contraste, ecotype…) sans qu’il soit nécéssaire de le subdiviser.</p>
<p>###Analyse des résultats POD:
Les fichiers résultats _*mat_omega_, _*summary_pi_xtx.out_, sont copiés tel quel en local, le fichier _*summary_contrast.out_, si il contient les résultats de plusieurs combinaison de contrastes, doit être subdivisé par combinaison au préalable et aussi copiées en local.
La matrice Ω POD est comparée à la matrice Ω initialement calculée (omegaB) afin de valider la similarité des analyses. La fonction quantile calcule le seuil en fonction de la valeur probs qu’on lui donne : probs = 0,99 pour un seuil à 1%, probs=0,999 pour un seuil à 0,1% etc.
Ces seuils seront utilisé comme seuil de significativité pour le jeu de données réel.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="annexes.html#cb45-1" tabindex="-1"></a>POD.omega<span class="ot">=</span><span class="fu">as.matrix</span>(<span class="fu">read.table</span>(<span class="fu">paste</span>(path_POD, <span class="st">&quot;project.POD_mat_omega.out&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)))</span>
<span id="cb45-2"><a href="annexes.html#cb45-2" tabindex="-1"></a><span class="fu">plot</span>(POD.omega,omegaB) ; <span class="fu">abline</span>(<span class="at">a=</span><span class="dv">0</span>,<span class="at">b=</span><span class="dv">1</span>)</span>
<span id="cb45-3"><a href="annexes.html#cb45-3" tabindex="-1"></a>FMD.POD <span class="ot">&lt;-</span> <span class="fu">fmd.dist</span>(POD.omega,omegaB)</span>
<span id="cb45-4"><a href="annexes.html#cb45-4" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Distance FMD =&quot;</span>, FMD.POD, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb45-5"><a href="annexes.html#cb45-5" tabindex="-1"></a></span>
<span id="cb45-6"><a href="annexes.html#cb45-6" tabindex="-1"></a><span class="co">#Extrait la colonne des XtX et calcule un seuil correspondant au quantile que l&#39;on souhaite</span></span>
<span id="cb45-7"><a href="annexes.html#cb45-7" tabindex="-1"></a>POD.XtX<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste</span>(path_POD, <span class="st">&quot;project.POD_summary_pi_xtx.out&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),<span class="at">h=</span>T)<span class="sc">$</span>M_XtX</span>
<span id="cb45-8"><a href="annexes.html#cb45-8" tabindex="-1"></a>thresh.XtX<span class="ot">=</span><span class="fu">quantile</span>(POD.XtX,<span class="at">probs=</span><span class="fl">0.99</span>)</span>
<span id="cb45-9"><a href="annexes.html#cb45-9" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Seuil XtX =&quot;</span>, thresh.XtX, <span class="st">&quot;(Max=&quot;</span>, <span class="fu">max</span>(POD.XtX) ,<span class="st">&quot;)&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb45-10"><a href="annexes.html#cb45-10" tabindex="-1"></a></span>
<span id="cb45-11"><a href="annexes.html#cb45-11" tabindex="-1"></a><span class="co">#Extrait la colonne des C2 et calcule un seuil correspondant au quantile que l&#39;on souhaite</span></span>
<span id="cb45-12"><a href="annexes.html#cb45-12" tabindex="-1"></a>POD.C2<span class="ot">=</span><span class="fu">read.table</span>(<span class="fu">paste</span>(path_POD, <span class="st">&quot;project.POD_summary_contrast.out&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),<span class="at">h=</span>T)<span class="sc">$</span>M_C2</span>
<span id="cb45-13"><a href="annexes.html#cb45-13" tabindex="-1"></a>thresh.C2<span class="ot">=</span><span class="fu">quantile</span>(POD.C2, <span class="at">probs=</span><span class="fl">0.99</span>)</span>
<span id="cb45-14"><a href="annexes.html#cb45-14" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;Seuil de contraste C2 = &quot;</span>, thresh.C2, <span class="st">&quot;(Max=&quot;</span>, <span class="fu">max</span>(POD.C2) ,<span class="st">&quot;)&quot;</span>, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="approche-par-fenêtres-glissantes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="références.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["BayPass_Fr.pdf", "BayPass_Fr.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
